<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">apaQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https/github.com/brimittleman/apaQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-09-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>apaQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanapaQTLtreef8cb7b8003863aa0430bce2d14dfe916c8c74c2atargetblankf8cb7b8a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/apaQTL/tree/f8cb7b8003863aa0430bce2d14dfe916c8c74c2a" target="_blank">f8cb7b8</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanapaQTLtreef8cb7b8003863aa0430bce2d14dfe916c8c74c2atargetblankf8cb7b8a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    docs/.DS_Store
    Ignored:    docs/figure/.DS_Store
    Ignored:    docs/figure/PAS_graphs_total.Rmd/.DS_Store
    Ignored:    docs/figure/choosePCs.Rmd/.DS_Store
    Ignored:    docs/figure/exvunexpeQTL.Rmd/.DS_Store
    Ignored:    docs/figure/snpinSS.Rmd/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  .Rprofile
    Untracked:  ._.DS_Store
    Untracked:  .gitignore
    Untracked:  @
    Untracked:  GEO_brimittleman/
    Untracked:  _workflowr.yml
    Untracked:  analysis/._PASdescriptiveplots.Rmd
    Untracked:  analysis/._cuttoffPercUsage.Rmd
    Untracked:  analysis/APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz.qqnorm.allChrom
    Untracked:  analysis/APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz.qqnorm.allChrom
    Untracked:  analysis/QTLexampleplots.Rmd
    Untracked:  analysis/cuttoffPercUsage.Rmd
    Untracked:  analysis/eQTLoverlap.Rmd
    Untracked:  analysis/interpret verify bam.Rmd
    Untracked:  analysis/interpret_verifybam.Rmd
    Untracked:  analysis/mergeRNA.Rmd
    Untracked:  analysis/oldstuffNotNeeded.Rmd
    Untracked:  analysis/remove_badlines.Rmd
    Untracked:  analysis/totSpecInNuclear.Rmd
    Untracked:  analysis/totSpecIncludenotTested.Rmd
    Untracked:  analysis/totalspec.Rmd
    Untracked:  apaQTL.Rproj
    Untracked:  checksumsfastq.txt.gz
    Untracked:  code/.NascentRNAdtPlotFirstintronicPAS.sh.swp
    Untracked:  code/._ApaQTL_nominalNonnorm.sh
    Untracked:  code/._BothFracDTPlotGeneRegions.sh
    Untracked:  code/._BothFracDTPlotGeneRegions_normalized.sh
    Untracked:  code/._DistPAS2Sig_RandomIntron.py
    Untracked:  code/._EandPqtl_perm.sh
    Untracked:  code/._EandPqtls.sh
    Untracked:  code/._FC_NucintornUpandDown.sh
    Untracked:  code/._FC_UTR.sh
    Untracked:  code/._FC_intornUpandDownsteamPAS.sh
    Untracked:  code/._FC_nascentseq.sh
    Untracked:  code/._FC_newPeaks_olddata.sh
    Untracked:  code/._HMMpermuteTotal.py
    Untracked:  code/._HmmPermute.py
    Untracked:  code/._IntronicPASDT.sh
    Untracked:  code/._LC_samplegroups.py
    Untracked:  code/._LD_qtl.sh
    Untracked:  code/._LD_snpsproxy.sh
    Untracked:  code/._NascentRNAdtPlot.sh
    Untracked:  code/._NascentRNAdtPlot3UTRPAS.sh
    Untracked:  code/._NascentRNAdtPlotExcludeFirstintronicPAS.sh
    Untracked:  code/._NascentRNAdtPlotNucPAS.sh
    Untracked:  code/._NascentRNAdtPlotTotPAS.sh
    Untracked:  code/._NascentRNAdtPlotintronicPAS.sh
    Untracked:  code/._NascnetRNAdtPlotPAS.sh
    Untracked:  code/._NetSeq_fourthintronDT.sh
    Untracked:  code/._NomResfromPASSNP.py
    Untracked:  code/._NuclearPAS_5per.bed.py
    Untracked:  code/._PTTfacetboxplots.R
    Untracked:  code/._PrematureQTLNominal.sh
    Untracked:  code/._PrematureQTLPermuted.sh
    Untracked:  code/._QTL2bed.py
    Untracked:  code/._QTL2bed_withstrand.py
    Untracked:  code/._RNAbam2bw.sh
    Untracked:  code/._RNAseqDTplot.sh
    Untracked:  code/._RunRes2PAS.sh
    Untracked:  code/._SAF215upbed.py
    Untracked:  code/._SnakefilePAS
    Untracked:  code/._SnakefilefiltPAS
    Untracked:  code/._TESplots100bp.sh
    Untracked:  code/._TESplots150bp.sh
    Untracked:  code/._TESplots200bp.sh
    Untracked:  code/._TotalPAS_5perc.bed.py
    Untracked:  code/._Untitled
    Untracked:  code/._ZipandTabPheno.sh
    Untracked:  code/._aAPAqtl_nominal39ind.sh
    Untracked:  code/._allNucSpecQTLine.py
    Untracked:  code/._allNucSpecfromNonNorm.py
    Untracked:  code/._annotatePacBioPASregion.sh
    Untracked:  code/._annotatedPAS2bed.py
    Untracked:  code/._apaInPandE.py
    Untracked:  code/._apaQTLCorrectPvalMakeQQ.R
    Untracked:  code/._apaQTLCorrectpval_6or7a.R
    Untracked:  code/._apaQTL_Nominal.sh
    Untracked:  code/._apaQTL_nominalInclusive.sh
    Untracked:  code/._apaQTL_nominalv67.sh
    Untracked:  code/._apaQTL_permuted.sh
    Untracked:  code/._apaQTL_permuted_test6A7A.sh
    Untracked:  code/._apainRibo.py
    Untracked:  code/._assignNucIntonpeak2intronlocs.sh
    Untracked:  code/._assignTotIntronpeak2intronlocs.sh
    Untracked:  code/._bam2BW_5primemost.sh
    Untracked:  code/._bed2saf.py
    Untracked:  code/._bothFracDTplot1stintron.sh
    Untracked:  code/._bothFracDTplot4thintron.sh
    Untracked:  code/._bothFrac_FC.sh
    Untracked:  code/._callPeaksYL.py
    Untracked:  code/._changeRibonomQTLres2genename.py
    Untracked:  code/._changenomQTLres2geneName.py
    Untracked:  code/._chooseAnno2PAS_pacbio.py
    Untracked:  code/._chooseAnno2SAF.py
    Untracked:  code/._chooseSignalSite
    Untracked:  code/._chooseSignalSite.py
    Untracked:  code/._closestannotated.sh
    Untracked:  code/._closestannotated_byfrac.sh
    Untracked:  code/._cluster.json
    Untracked:  code/._clusterPAS.json
    Untracked:  code/._clusterfiltPAS.json
    Untracked:  code/._codingdms2bed.py
    Untracked:  code/._config.yaml
    Untracked:  code/._config2.yaml
    Untracked:  code/._configOLD.yaml
    Untracked:  code/._convertNominal2SNPLOC.py
    Untracked:  code/._convertNominal2SNPloc2Versions.py
    Untracked:  code/._convertNumeric.py
    Untracked:  code/._correctNomeqtl.R
    Untracked:  code/._createPlinkSampfile.py
    Untracked:  code/._dag.pdf
    Untracked:  code/._eQTL_switch2snploc.py
    Untracked:  code/._eQTLgenestestedapa.py
    Untracked:  code/._encodeRNADTplots.sh
    Untracked:  code/._extractGenotypes.py
    Untracked:  code/._extractseqfromqtlfastq.py
    Untracked:  code/._fc2leafphen.py
    Untracked:  code/._fc_filteredPAS6and7As.sh
    Untracked:  code/._fifteenBPupstreamPAS.py
    Untracked:  code/._fiftyBPupstreamPAS.py
    Untracked:  code/._filter5perc.R
    Untracked:  code/._filter5percPheno.py
    Untracked:  code/._filterLDsnps.py
    Untracked:  code/._filterMPPAS.py
    Untracked:  code/._filterMPPAS_15.py
    Untracked:  code/._filterMPPAS_15_7As.py
    Untracked:  code/._filterMPPAS_50.py
    Untracked:  code/._filterSAFforMP.py
    Untracked:  code/._filterpeaks.py
    Untracked:  code/._finalPASbed2SAF.py
    Untracked:  code/._fix4su304corr.py
    Untracked:  code/._fix4su604corr.py
    Untracked:  code/._fix4sukalisto.py
    Untracked:  code/._fixExandUnexeQTL
    Untracked:  code/._fixExandUnexeQTL.py
    Untracked:  code/._fixFChead.py
    Untracked:  code/._fixFChead_bothfrac.py
    Untracked:  code/._fixFChead_short.py
    Untracked:  code/._fixH3k12ac.py
    Untracked:  code/._fixPASregionSNPs.py
    Untracked:  code/._fixRNAhead4corr.py
    Untracked:  code/._fixRNAkalisto.py
    Untracked:  code/._fix_randomIntron.py
    Untracked:  code/._fixgroupedtranscript.py
    Untracked:  code/._fixhead_netseqfc.py
    Untracked:  code/._getAPAfromanyeQTL.py
    Untracked:  code/._getApapval4eqtl.py
    Untracked:  code/._getApapval4eqtl_unexp.py
    Untracked:  code/._getApapval4eqtl_version67.py
    Untracked:  code/._getDownstreamIntronNuclear.py
    Untracked:  code/._getIntronDownstreamPAS.py
    Untracked:  code/._getIntronUpstreamPAS.py
    Untracked:  code/._getQTLalleles.py
    Untracked:  code/._getQTLfastq.sh
    Untracked:  code/._getUpstreamIntronNuclear.py
    Untracked:  code/._grouptranscripts.py
    Untracked:  code/._intersectVCFandupPAS.sh
    Untracked:  code/._keep5perMAF.py
    Untracked:  code/._keepSNP_vcf.sh
    Untracked:  code/._make5percPeakbed.py
    Untracked:  code/._makeFileID.py
    Untracked:  code/._makePheno.py
    Untracked:  code/._makeSAFbothfrac5perc.py
    Untracked:  code/._makeSNP2rsidfile.py
    Untracked:  code/._makeeQTLempirical_unexp.py
    Untracked:  code/._makeeQTLempiricaldist.py
    Untracked:  code/._makegencondeTSSfile.py
    Untracked:  code/._mapSSsnps2PAS.sh
    Untracked:  code/._mergRNABam.sh
    Untracked:  code/._mergeAllBam.sh
    Untracked:  code/._mergeAnnotations.sh
    Untracked:  code/._mergeBW_norm.sh
    Untracked:  code/._mergeBamNascent.sh
    Untracked:  code/._mergeByFracBam.sh
    Untracked:  code/._mergePeaks.sh
    Untracked:  code/._mnase1stintron.sh
    Untracked:  code/._mnaseDT_fourthintron.sh
    Untracked:  code/._namePeaks.py
    Untracked:  code/._netseqDTplot1stIntron.sh
    Untracked:  code/._netseqFC.sh
    Untracked:  code/._nucQTLGWAS.py
    Untracked:  code/._nucSpecQTLineData.py
    Untracked:  code/._nucSpeceffectsize.py
    Untracked:  code/._nucspecnucPASine.py
    Untracked:  code/._pQTLsotherdata.py
    Untracked:  code/._pacbioDT.sh
    Untracked:  code/._pacbioIntronicDT.sh
    Untracked:  code/._parseBestbamid.py
    Untracked:  code/._peak2PAS.py
    Untracked:  code/._peakFC.sh
    Untracked:  code/._pheno2countonly.R
    Untracked:  code/._phenoQTLfromlist.py
    Untracked:  code/._processYRIgen.py
    Untracked:  code/._pttQTLsinapaQTL.py
    Untracked:  code/._qtlRegionseq.sh
    Untracked:  code/._qtlsPvalOppFrac.py
    Untracked:  code/._quantassign2parsedpeak.py
    Untracked:  code/._removeXfromHmm.py
    Untracked:  code/._removeloc_pheno.py
    Untracked:  code/._riboQTL.sh
    Untracked:  code/._runCorrectNomEqtl.sh
    Untracked:  code/._runHMMpermuteAPAqtls.sh
    Untracked:  code/._runHMMpermuteeQTLS.sh
    Untracked:  code/._runMakeEmpiricaleQTL_unexp.sh
    Untracked:  code/._runMakeeQTLempirical.sh
    Untracked:  code/._run_bam2bw_all3prime.sh
    Untracked:  code/._run_bam2bw_extra3.sh
    Untracked:  code/._run_bestbamid.sj
    Untracked:  code/._run_dist2sig_randomintron.sh
    Untracked:  code/._run_filtersnpLD.sh
    Untracked:  code/._run_getAPAfromeQTL_version6.7.sh
    Untracked:  code/._run_getApaPval4eqtl.sh
    Untracked:  code/._run_getapafromeQTL.py
    Untracked:  code/._run_getapafromeQTL.sh
    Untracked:  code/._run_getapapval4eqtl_unexp.sh
    Untracked:  code/._run_leafcutterDiffIso.sh
    Untracked:  code/._run_prxySNP.sh
    Untracked:  code/._run_pttfacetboxplot.sh
    Untracked:  code/._run_sepUsagephen.sh
    Untracked:  code/._run_sepgenobychrom.sh
    Untracked:  code/._run_verifybam.sh
    Untracked:  code/._selectNominalPvalues.py
    Untracked:  code/._sepUsagePhen.py
    Untracked:  code/._sepgenobychrom.py
    Untracked:  code/._snakemakePAS.batch
    Untracked:  code/._snakemakefiltPAS.batch
    Untracked:  code/._sortindexRNAbam.sh
    Untracked:  code/._specAPAinE.py
    Untracked:  code/._submit-snakemakePAS.sh
    Untracked:  code/._submit-snakemakefiltPAS.sh
    Untracked:  code/._subsetAPAnotEorPgene.py
    Untracked:  code/._subsetAPAnotEorPgene_2versions.py
    Untracked:  code/._subsetApanoteGene.py
    Untracked:  code/._subsetApanoteGene_2versions.py
    Untracked:  code/._subsetUnexplainedeQTLs.py
    Untracked:  code/._subsetVCF_SS.sh
    Untracked:  code/._subsetVCF_noSSregions.sh
    Untracked:  code/._subsetVCF_upstreamPAS.sh
    Untracked:  code/._subset_diffisopheno.py
    Untracked:  code/._subsetpermAPAwithGenelist.py
    Untracked:  code/._subsetpermAPAwithGenelist_2versions.py
    Untracked:  code/._subsetvcf_otherreg.sh
    Untracked:  code/._subsetvcf_permSS.sh
    Untracked:  code/._subtrachfiveprimeUTR.sh
    Untracked:  code/._subtractExons.sh
    Untracked:  code/._subtractfiveprimeUTR.sh
    Untracked:  code/._tabixSNPS.sh
    Untracked:  code/._tenBPupstreamPAS.py
    Untracked:  code/._testVerifyBam.sh
    Untracked:  code/._totSeceffectsize.py
    Untracked:  code/._twentyBPupstreamPAS.py
    Untracked:  code/._utrdms2saf.py
    Untracked:  code/._vcf2bed.py
    Untracked:  code/._verifyBam18517N.sh
    Untracked:  code/._verifyBam18517T.sh
    Untracked:  code/._verifyBam19128N.sh
    Untracked:  code/._verifyBam19128T.sh
    Untracked:  code/._wrap_verifybam.sh
    Untracked:  code/._writePTTexamplecode.py
    Untracked:  code/._writePTTexamplecode.sh
    Untracked:  code/.pversion
    Untracked:  code/.snakemake/
    Untracked:  code/1
    Untracked:  code/APAqtl_nominal.err
    Untracked:  code/APAqtl_nominal.out
    Untracked:  code/APAqtl_nominal_39.err
    Untracked:  code/APAqtl_nominal_39.out
    Untracked:  code/APAqtl_nominal_inclusive.err
    Untracked:  code/APAqtl_nominal_inclusive.out
    Untracked:  code/APAqtl_nominal_nonNorm.err
    Untracked:  code/APAqtl_nominal_nonNorm.out
    Untracked:  code/APAqtl_nominal_versions67.err
    Untracked:  code/APAqtl_nominal_versions67.out
    Untracked:  code/APAqtl_permuted.err
    Untracked:  code/APAqtl_permuted.out
    Untracked:  code/APAqtl_permuted_versions67.err
    Untracked:  code/APAqtl_permuted_versions67.out
    Untracked:  code/BothFracDTPlot1stintron.err
    Untracked:  code/BothFracDTPlot1stintron.out
    Untracked:  code/BothFracDTPlot4stintron.err
    Untracked:  code/BothFracDTPlot4stintron.out
    Untracked:  code/BothFracDTPlotGeneRegions.err
    Untracked:  code/BothFracDTPlotGeneRegions.out
    Untracked:  code/BothFracDTPlotGeneRegions_norm.err
    Untracked:  code/BothFracDTPlotGeneRegions_norm.out
    Untracked:  code/DistPAS2Sig_RandomIntron.py
    Untracked:  code/EandPqtl.err
    Untracked:  code/EandPqtl.out
    Untracked:  code/EncodeRNADTPlotGeneRegions.err
    Untracked:  code/EncodeRNADTPlotGeneRegions.out
    Untracked:  code/FC_NucintronPASupandDown.err
    Untracked:  code/FC_NucintronPASupandDown.out
    Untracked:  code/FC_UTR.err
    Untracked:  code/FC_UTR.out
    Untracked:  code/FC_intronPASupandDown.err
    Untracked:  code/FC_intronPASupandDown.out
    Untracked:  code/FC_nascent.err
    Untracked:  code/FC_nascentout
    Untracked:  code/FC_newPAS_olddata.err
    Untracked:  code/FC_newPAS_olddata.out
    Untracked:  code/HmmPermute.p
    Untracked:  code/IntronicPASDT.err
    Untracked:  code/IntronicPASDT.out
    Untracked:  code/LD_vcftools.hap.out
    Untracked:  code/NascentDTPlotGeneRegions.err
    Untracked:  code/NascentDTPlotGeneRegions.out
    Untracked:  code/NascentDTPlotPAS.err
    Untracked:  code/NascentDTPlotPAS.out
    Untracked:  code/NascentDTPlotPAS_3utr.err
    Untracked:  code/NascentDTPlotPAS_3utr.out
    Untracked:  code/NascentDTPlotPAS_firstintron.err
    Untracked:  code/NascentDTPlotPAS_firstintron.out
    Untracked:  code/NascentDTPlotPAS_intron.err
    Untracked:  code/NascentDTPlotPAS_intron.out
    Untracked:  code/NascentDTPlotPAS_nuc.err
    Untracked:  code/NascentDTPlotPAS_nuc.out
    Untracked:  code/NascentDTPlotPAS_tot.err
    Untracked:  code/NascentDTPlotPAS_tot.out
    Untracked:  code/Nuclear_example.err
    Untracked:  code/Nuclear_example.out
    Untracked:  code/PACbioDT.err
    Untracked:  code/PACbioDT.out
    Untracked:  code/PACbioDTitronic.err
    Untracked:  code/PACbioDTitronic.out
    Untracked:  code/Prematureqtl_nominal.err
    Untracked:  code/Prematureqtl_nominal.out
    Untracked:  code/Prematureqtl_permuted.err
    Untracked:  code/Prematureqtl_permuted.out
    Untracked:  code/README.md
    Untracked:  code/RNABam2BW.err
    Untracked:  code/RNABam2BW.out
    Untracked:  code/RNAseqDTPlotGeneRegions.err
    Untracked:  code/RNAseqDTPlotGeneRegions.out
    Untracked:  code/Rplots.pdf
    Untracked:  code/TESplots100bp.err
    Untracked:  code/TESplots100bp.out
    Untracked:  code/TESplots150bp.err
    Untracked:  code/TESplots150bp.out
    Untracked:  code/TESplots200bp.err
    Untracked:  code/TESplots200bp.out
    Untracked:  code/Total_example.err
    Untracked:  code/Total_example.out
    Untracked:  code/Untitled
    Untracked:  code/YRI_LCL.vcf.gz
    Untracked:  code/YRI_LCL_chr1.vcf.gz.log
    Untracked:  code/YRI_LCL_chr1.vcf.gz.recode.vcf
    Untracked:  code/annotatedPASregion.err
    Untracked:  code/annotatedPASregion.out
    Untracked:  code/apaQTL_nominalInclusive.sh
    Untracked:  code/assignPeak2Intronicregion.err
    Untracked:  code/assignPeak2Intronicregion.out
    Untracked:  code/assigntotPeak2Intronicregion.err
    Untracked:  code/assigntotPeak2Intronicregion.out
    Untracked:  code/bam2bw.err
    Untracked:  code/bam2bw.out
    Untracked:  code/bam2bw_5primemost.err
    Untracked:  code/bam2bw_5primemost.out
    Untracked:  code/binary_fileset.log
    Untracked:  code/bothFrac_FC.err
    Untracked:  code/bothFrac_FC.out
    Untracked:  code/callSHscripts.txt
    Untracked:  code/closestannotated.err
    Untracked:  code/closestannotated.out
    Untracked:  code/closestannotatedbyfrac.err
    Untracked:  code/closestannotatedbyfrac.out
    Untracked:  code/dag.pdf
    Untracked:  code/dagPAS.pdf
    Untracked:  code/dagfiltPAS.pdf
    Untracked:  code/fixExandUnexeQTL
    Untracked:  code/fix_randomIntron.py
    Untracked:  code/genotypesYRI.gen.proc.keep.vcf.log
    Untracked:  code/genotypesYRI.gen.proc.keep.vcf.recode.vcf
    Untracked:  code/getseq100up.err
    Untracked:  code/getseq100up.out
    Untracked:  code/grouptranscripts.err
    Untracked:  code/grouptranscripts.out
    Untracked:  code/intersectPAS_ssSNPS.err
    Untracked:  code/intersectPAS_ssSNPS.out
    Untracked:  code/intersectVCFPAS.err
    Untracked:  code/intersectVCFPAS.out
    Untracked:  code/log/
    Untracked:  code/mergeAnno.err
    Untracked:  code/mergeAnno.out
    Untracked:  code/mergeBWnorm.err
    Untracked:  code/mergeBWnorm.out
    Untracked:  code/mergeBamNacent.err
    Untracked:  code/mergeBamNacent.out
    Untracked:  code/mergeRNAbam.err
    Untracked:  code/mergeRNAbam.out
    Untracked:  code/mnaseDTPlot1stintron.err
    Untracked:  code/mnaseDTPlot1stintron.out
    Untracked:  code/mnaseDTPlot4thintron.err
    Untracked:  code/mnaseDTPlot4thintron.out
    Untracked:  code/netDTPlot4thintron.out
    Untracked:  code/netseqFC.err
    Untracked:  code/netseqFC.out
    Untracked:  code/neyDTPlot4thintron.err
    Untracked:  code/nucspecinE.py
    Untracked:  code/plink.log
    Untracked:  code/prxySNP.err
    Untracked:  code/prxySNP.out
    Untracked:  code/pttFacetBoxplots.err
    Untracked:  code/pttFacetBoxplots.out
    Untracked:  code/qtlFacetBoxplots.err
    Untracked:  code/qtlFacetBoxplots.out
    Untracked:  code/rLD_vcftools.hap.err
    Untracked:  code/riboqtl.err
    Untracked:  code/riboqtl.out
    Untracked:  code/runBestBamID.err
    Untracked:  code/runCorrectNomeqtl.err
    Untracked:  code/runCorrectNomeqtl.out
    Untracked:  code/runFilterLD.err
    Untracked:  code/runFilterLD.out
    Untracked:  code/runHMMpermute.err
    Untracked:  code/runHMMpermute.out
    Untracked:  code/runHMMpermuteeQTLs.err
    Untracked:  code/runHMMpermuteeQTLs.out
    Untracked:  code/runMakeEmpiricaleQTLs.err
    Untracked:  code/runMakeEmpiricaleQTLs.out
    Untracked:  code/runMakeEmpiricaleQTLsunex.err
    Untracked:  code/runMakeEmpiricaleQTLsunex.out
    Untracked:  code/run_DistPAS2Sig.err
    Untracked:  code/run_DistPAS2Sig.out
    Untracked:  code/run_DistPAS2Sig_intron.err
    Untracked:  code/run_DistPAS2Sig_intron.out
    Untracked:  code/run_bam2bw.err
    Untracked:  code/run_bam2bw.out
    Untracked:  code/run_bam2bwexta.err
    Untracked:  code/run_bam2bwexta.out
    Untracked:  code/run_dist2sig_randomintron.sh
    Untracked:  code/run_getAPAfromanyeQTL.err
    Untracked:  code/run_getAPAfromanyeQTL.out
    Untracked:  code/run_getApaPval4eQTLs.err
    Untracked:  code/run_getApaPval4eQTLs.out
    Untracked:  code/run_getApaPval4eQTLsunexplained.err
    Untracked:  code/run_getApaPval4eQTLsunexplained.out
    Untracked:  code/run_leafcutter_ds.err
    Untracked:  code/run_leafcutter_ds.out
    Untracked:  code/run_sepgenobychrom.err
    Untracked:  code/run_sepgenobychrom.out
    Untracked:  code/run_sepusage.err
    Untracked:  code/run_sepusage.out
    Untracked:  code/run_verifybam.err
    Untracked:  code/run_verifybam.out
    Untracked:  code/run_verifybam128N.err
    Untracked:  code/run_verifybam128N.out
    Untracked:  code/run_verifybam128T.err
    Untracked:  code/run_verifybam128T.out
    Untracked:  code/run_verifybam517N.err
    Untracked:  code/run_verifybam517N.out
    Untracked:  code/run_verifybam517T.err
    Untracked:  code/run_verifybam517T.out
    Untracked:  code/runprxySNP.err
    Untracked:  code/runprxySNP.out
    Untracked:  code/runres2pas.err
    Untracked:  code/runres2pas.out
    Untracked:  code/seqQTLfastq.err
    Untracked:  code/seqQTLfastq.out
    Untracked:  code/seqQTLregion.err
    Untracked:  code/seqQTLregion.out
    Untracked:  code/snakePASlog.out
    Untracked:  code/snakefiltPASlog.out
    Untracked:  code/sortindexRNABam.err
    Untracked:  code/sortindexRNABam.out
    Untracked:  code/specAPAinE.py
    Untracked:  code/subsetvcf_SS.err
    Untracked:  code/subsetvcf_SS.out
    Untracked:  code/subsetvcf_noSS.err
    Untracked:  code/subsetvcf_noSS.out
    Untracked:  code/subsetvcf_pas.err
    Untracked:  code/subsetvcf_pas.out
    Untracked:  code/subsetvcf_perm.err
    Untracked:  code/subsetvcf_perm.out
    Untracked:  code/subsetvcf_rand.err
    Untracked:  code/subsetvcf_rand.out
    Untracked:  code/subtract5UTR.err
    Untracked:  code/subtract5UTR.out
    Untracked:  code/subtractExons.err
    Untracked:  code/subtractExons.out
    Untracked:  code/tabixSNPs.err
    Untracked:  code/tabixSNPs.out
    Untracked:  code/test_verifybam.err
    Untracked:  code/test_verifybam.out
    Untracked:  code/vcf_keepsnps.err
    Untracked:  code/vcf_keepsnps.out
    Untracked:  code/wrap_verifybam.err
    Untracked:  code/wrap_verifybam.out
    Untracked:  code/zipandtabPhen.err
    Untracked:  code/zipandtabPhen.out
    Untracked:  data/._.DS_Store
    Untracked:  data/._MetaDataSequencing.txt
    Untracked:  data/AnnotatedPAS/
    Untracked:  data/ApaByEgene/
    Untracked:  data/ApaByPgene/
    Untracked:  data/BadLines/
    Untracked:  data/Battle_pQTL/
    Untracked:  data/CheckSums/
    Untracked:  data/CompareOldandNew/
    Untracked:  data/DTmatrix/
    Untracked:  data/DiffIso/
    Untracked:  data/EncodeRNA/
    Untracked:  data/ExampleQTLPlots/
    Untracked:  data/ExpressionIndependentapaQTLs.txt
    Untracked:  data/FlaggedPAS/
    Untracked:  data/GWAS_overlap/
    Untracked:  data/GeuvadisRNA/
    Untracked:  data/HMMqtls/
    Untracked:  data/Li_eQTLs/
    Untracked:  data/NascentRNA/
    Untracked:  data/NucSpeceQTLeffect/
    Untracked:  data/PAS/
    Untracked:  data/PAS_postFlag/
    Untracked:  data/PolyA_DB/
    Untracked:  data/PreTerm_pheno/
    Untracked:  data/PrematureQTLNominal/
    Untracked:  data/PrematureQTLPermuted/
    Untracked:  data/QTLGenotypes/
    Untracked:  data/QTLoverlap/
    Untracked:  data/QTLoverlap_inclusive/
    Untracked:  data/QTLoverlap_nonNorm/
    Untracked:  data/README.md
    Untracked:  data/RNAseq/
    Untracked:  data/Reads2UTR/
    Untracked:  data/SNPinSS/
    Untracked:  data/SignalSiteFiles/
    Untracked:  data/TF_motifdisruption/
    Untracked:  data/ThirtyNineIndQtl_nominal/
    Untracked:  data/Version15bp6As/
    Untracked:  data/Version15bp7As/
    Untracked:  data/apaQTLNominal/
    Untracked:  data/apaQTLNominal_4pc/
    Untracked:  data/apaQTLNominal_inclusive/
    Untracked:  data/apaQTLPermuted/
    Untracked:  data/apaQTLPermuted_4pc/
    Untracked:  data/apaQTLs/
    Untracked:  data/assignedPeaks/
    Untracked:  data/assignedPeaks_15Up/
    Untracked:  data/bam/
    Untracked:  data/bam_clean/
    Untracked:  data/bam_waspfilt/
    Untracked:  data/bed_10up/
    Untracked:  data/bed_clean/
    Untracked:  data/bed_clean_sort/
    Untracked:  data/bed_waspfilter/
    Untracked:  data/bedsort_waspfilter/
    Untracked:  data/bothFrac_FC/
    Untracked:  data/bw/
    Untracked:  data/bw_norm/
    Untracked:  data/eQTLs/
    Untracked:  data/exampleQTLs/
    Untracked:  data/fastq/
    Untracked:  data/filterPeaks/
    Untracked:  data/fourSU/
    Untracked:  data/h3k27ac/
    Untracked:  data/highdiffsiggenes.txt
    Untracked:  data/inclusivePeaks/
    Untracked:  data/inclusivePeaks_FC/
    Untracked:  data/intronRNAratio/
    Untracked:  data/intron_analysis/
    Untracked:  data/locusZoom/
    Untracked:  data/mergedBG/
    Untracked:  data/mergedBW_byfrac/
    Untracked:  data/mergedBW_norm/
    Untracked:  data/mergedBam/
    Untracked:  data/mergedbyFracBam/
    Untracked:  data/molPhenos/
    Untracked:  data/molQTLs/
    Untracked:  data/motifdistrupt/
    Untracked:  data/netseq/
    Untracked:  data/nonNorm_pheno/
    Untracked:  data/nuc_10up/
    Untracked:  data/nuc_10upclean/
    Untracked:  data/oldPASfiles/
    Untracked:  data/overlapeQTL_try2/
    Untracked:  data/overlapeQTLs/
    Untracked:  data/pQTLoverlap/
    Untracked:  data/pacbio/
    Untracked:  data/peakCoverage/
    Untracked:  data/peaks_5perc/
    Untracked:  data/phenotype/
    Untracked:  data/phenotype_5perc/
    Untracked:  data/phenotype_inclusivePAS/
    Untracked:  data/pttQTL/
    Untracked:  data/pttQTLplots/
    Untracked:  data/sigDiffGenes.txt
    Untracked:  data/sort/
    Untracked:  data/sort_clean/
    Untracked:  data/sort_waspfilter/
    Untracked:  data/twoMech/
    Untracked:  data/verifyBAM/
    Untracked:  data/verifyBAM_full/
    Untracked:  docs/._.DS_Store
    Untracked:  docs/figure/._.DS_Store
    Untracked:  docs/figure/PAS_graphs_total.Rmd/._.DS_Store
    Untracked:  docs/figure/choosePCs.Rmd/._.DS_Store
    Untracked:  docs/figure/exvunexpeQTL.Rmd/._.DS_Store
    Untracked:  docs/figure/snpinSS.Rmd/._.DS_Store
    Untracked:  nohup.out
    Untracked:  output/._.DS_Store
    Untracked:  output/._meanCorrelationPhenotypes.svg
    Untracked:  output/dtPlots/
    Untracked:  output/fastqc/
    Untracked:  output/meanCorrelationPhenotypes.svg
    Untracked:  run_verifybam517N.err
    Untracked:  run_verifybam517N.out

Unstaged changes:
    Modified:   analysis/NuclearSpecIncludeNotTested.Rmd
    Modified:   analysis/PASdescriptiveplots.Rmd
    Modified:   analysis/Readdistagainstfeatures.Rmd
    Modified:   analysis/compareAnnotatedpas.Rmd
    Modified:   analysis/nucSpecinEQTLs.Rmd
    Modified:   analysis/overlapapaqtlsandeqtls.Rmd
    Modified:   analysis/pQTLexampleplot.Rmd
    Modified:   analysis/propeQTLs_explained.Rmd
    Modified:   analysis/version15bpfilter.Rmd
    Modified:   code/DistPAS2Sig.py
    Modified:   code/apaQTLsnake.err
    Deleted:    code/test.txt
    Modified:   docs/figure/DiffIsoAnalysis.Rmd/figure1EHistogramversion-1.pdf
    Modified:   docs/figure/DiffIsoAnalysis.Rmd/figure1Emain-1.pdf
    Modified:   docs/figure/DiffIsoAnalysis.Rmd/figure1Esubset-1.pdf
    Modified:   docs/figure/signalsiteanalysis.Rmd/figure1bMain-1.pdf
    Deleted:    reads_graphs.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/f8cb7b8003863aa0430bce2d14dfe916c8c74c2a/analysis/index.Rmd" target="_blank">f8cb7b8</a>
</td>
<td>
brimittleman
</td>
<td>
2019-09-17
</td>
<td>
move inclusive, get numbers for paper
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/2bc41872f9000e1832157dbd2dbae24530238fa8/docs/index.html" target="_blank">2bc4187</a>
</td>
<td>
brimittleman
</td>
<td>
2019-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/bb2fbab1ba4b4312459e835a950d81b1a6d4eaec/docs/index.html" target="_blank">bb2fbab</a>
</td>
<td>
brimittleman
</td>
<td>
2019-09-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/42f352c8c8394aaea38999d9d6bbd876aaedb026/docs/index.html" target="_blank">42f352c</a>
</td>
<td>
brimittleman
</td>
<td>
2019-09-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/c5c6546e970a7e5ba37fe2fcf641883124ca7cfe/analysis/index.Rmd" target="_blank">c5c6546</a>
</td>
<td>
brimittleman
</td>
<td>
2019-09-07
</td>
<td>
add enrichment analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/3b2586048c7319f136082026b6bc3b260970d423/docs/index.html" target="_blank">3b25860</a>
</td>
<td>
brimittleman
</td>
<td>
2019-09-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5984fad7f726f1327bb79690f1598e54e9541570/analysis/index.Rmd" target="_blank">5984fad</a>
</td>
<td>
brimittleman
</td>
<td>
2019-09-04
</td>
<td>
update for new filter condition
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/07d9709bc73c4ce164b4364f22c11994976e898d/docs/index.html" target="_blank">07d9709</a>
</td>
<td>
brimittleman
</td>
<td>
2019-08-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/d5f19ccb09974ef7af14939dde3a2d8300a2542a/analysis/index.Rmd" target="_blank">d5f19cc</a>
</td>
<td>
brimittleman
</td>
<td>
2019-08-28
</td>
<td>
add new QC files to index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/4cd5f2813535e689adfa9c55d93d981267a23f70/docs/index.html" target="_blank">4cd5f28</a>
</td>
<td>
brimittleman
</td>
<td>
2019-08-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5289aa1cbd94dfc0369789c7078332339ed4edd0/analysis/index.Rmd" target="_blank">5289aa1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-08-26
</td>
<td>
add extra MP qc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/191d71f4b9e964948affa37dfb53778da65fed00/docs/index.html" target="_blank">191d71f</a>
</td>
<td>
brimittleman
</td>
<td>
2019-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/01d0be5fdfc21732bd65ef211ac2c35de4bc1831/analysis/index.Rmd" target="_blank">01d0be5</a>
</td>
<td>
brimittleman
</td>
<td>
2019-08-07
</td>
<td>
add reads_graph analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/d13025bc575c3e548de1520d9af6806a78512508/docs/index.html" target="_blank">d13025b</a>
</td>
<td>
brimittleman
</td>
<td>
2019-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/e2ff61a96cce652a6cf84cd2d16905e3ead802d9/analysis/index.Rmd" target="_blank">e2ff61a</a>
</td>
<td>
brimittleman
</td>
<td>
2019-07-31
</td>
<td>
add mayher plot to site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/33a700ae055a545ff1d3a8c2bf7d9d97f2e430b8/docs/index.html" target="_blank">33a700a</a>
</td>
<td>
brimittleman
</td>
<td>
2019-07-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/f3e3e163403506253b5b9b6c67aa6b519ed3bd53/analysis/index.Rmd" target="_blank">f3e3e16</a>
</td>
<td>
brimittleman
</td>
<td>
2019-07-25
</td>
<td>
small edits from paper writing and add gwas analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/fb1fde62ce14925599b84571616edfd5433fe318/docs/index.html" target="_blank">fb1fde6</a>
</td>
<td>
brimittleman
</td>
<td>
2019-07-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/2572d13fdfcffbc34a3ea77dee3701d41f942a0e/analysis/index.Rmd" target="_blank">2572d13</a>
</td>
<td>
brimittleman
</td>
<td>
2019-07-16
</td>
<td>
add compare annotated and additional coverage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/272b0b4f1aed7bbe9ee7048a006b30e874f3cfa1/docs/index.html" target="_blank">272b0b4</a>
</td>
<td>
brimittleman
</td>
<td>
2019-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/b1e6dd1873e3cc8bc0c05026a0ef4554c7d8819d/analysis/index.Rmd" target="_blank">b1e6dd1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-07-08
</td>
<td>
update ptt analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/5566d80d193fc0bc7a9c04534806fb842536d67f/docs/index.html" target="_blank">5566d80</a>
</td>
<td>
brimittleman
</td>
<td>
2019-07-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/f85f6d0a598ea970e3f520af9d1a75fee6e6abf4/analysis/index.Rmd" target="_blank">f85f6d0</a>
</td>
<td>
brimittleman
</td>
<td>
2019-07-07
</td>
<td>
add ptt eqtls
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/2fb467b818f6a78e6f061c1ba596a23c18cebf3a/docs/index.html" target="_blank">2fb467b</a>
</td>
<td>
brimittleman
</td>
<td>
2019-07-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/bd5f2284d2f758d77ef58a44838a562f0a0325fc/analysis/index.Rmd" target="_blank">bd5f228</a>
</td>
<td>
brimittleman
</td>
<td>
2019-07-07
</td>
<td>
add nuc spec eqtl res
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/a4a34bfa230453d858564379b37585646571d62b/docs/index.html" target="_blank">a4a34bf</a>
</td>
<td>
brimittleman
</td>
<td>
2019-07-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/5edd9c70f2339f870563a88f0c77a19f8ff6e7af/docs/index.html" target="_blank">5edd9c7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/709b4c88a3daf833806940eb0e35ef49d2bfc1c9/analysis/index.Rmd" target="_blank">709b4c8</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-30
</td>
<td>
add two reg
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/da5807e3529c3be37cb9591f3fbf151385d68220/docs/index.html" target="_blank">da5807e</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/1e58d078dffb6c8c07e901416dca941f773f4b75/docs/index.html" target="_blank">1e58d07</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/db5d2e6a82682bc8a7e0a381ff7cfbe6f88cd292/analysis/index.Rmd" target="_blank">db5d2e6</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-27
</td>
<td>
add pqtl example
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/5d97550add992d2169019ee4864f4a8dc75d26cf/docs/index.html" target="_blank">5d97550</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/3e288f8b007ba62da0ba1c289b26d071ca09a400/analysis/index.Rmd" target="_blank">3e288f8</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-25
</td>
<td>
add tf analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/433092b3ac3b77d6d551564291163697bc18d68b/docs/index.html" target="_blank">433092b</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/78dd5da969049a7c26369936aa70b58e53743881/analysis/index.Rmd" target="_blank">78dd5da</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-24
</td>
<td>
add snp in ss analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/fcab450e38a2ae2bbb8c9cfc254123cc1728fcc5/docs/index.html" target="_blank">fcab450</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5c834968e82472bde3d4116e2659e33a2abee99c/analysis/index.Rmd" target="_blank">5c83496</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-21
</td>
<td>
fix html typo
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/cd60f500464e8501b43d1fed4293158c01ce4a83/docs/index.html" target="_blank">cd60f50</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/6df08b6071d3f3b93b3ed3c759b424fc0967d976/analysis/index.Rmd" target="_blank">6df08b6</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-20
</td>
<td>
change analysis to include not tested in total as nuc spec
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/529a38a7ddc53d974f2dc20122d031d582118ee0/docs/index.html" target="_blank">529a38a</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/b3328b699912f8f0e3a7f1946927afd535d049ed/docs/index.html" target="_blank">b3328b6</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/01bc8aa8f4b26a38e94ed2db923aa0778295d3e8/analysis/index.Rmd" target="_blank">01bc8aa</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-18
</td>
<td>
add verify first inton res
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/743c9c6e421a227c55632ab6db2c67e658b6225c/docs/index.html" target="_blank">743c9c6</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/bc3ddae8692988ab00f61f8089953b7f540b55dc/analysis/index.Rmd" target="_blank">bc3ddae</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-17
</td>
<td>
add effect size nuc spec
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/2b8bf0c04f31d885fea508981e63eb5a3ad2204f/docs/index.html" target="_blank">2b8bf0c</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/54946d73b9da2c8e7eb7e8c2134096584c1809c6/analysis/index.Rmd" target="_blank">54946d7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-17
</td>
<td>
add mol qtl code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/44d1fcc7e477d470eda8201177061bf2ccad0d3e/docs/index.html" target="_blank">44d1fcc</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/a5430c340144fa66c73ca88f50441dd0a20609a9/analysis/index.Rmd" target="_blank">a5430c3</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-16
</td>
<td>
add qtl and usage analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/3fbf3f3041b64f6c84c1482517a619a0242e0ac6/docs/index.html" target="_blank">3fbf3f3</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/dc8d012e3b11bd95d50b9a461ef8b1ab8047eec7/analysis/index.Rmd" target="_blank">dc8d012</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-14
</td>
<td>
add nuclear specific analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/71ae7085008d874cce9074a415c8dedebd96cf4a/docs/index.html" target="_blank">71ae708</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/15962c894bfeb221c45d3f1260561c58a29c0941/docs/index.html" target="_blank">15962c8</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/b38964e1734f523e65ec48706cd046efad0babc5/analysis/index.Rmd" target="_blank">b38964e</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-14
</td>
<td>
add pqtl overlap
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/4912eaafa457d31e530e63d4ec4b38e26caca4ed/docs/index.html" target="_blank">4912eaa</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/bf91cb3d5abf46b7a92fd9201837557a27907cfb/analysis/index.Rmd" target="_blank">bf91cb3</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-14
</td>
<td>
add location plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/2c39fd1e839dc65e370efbdd5f23615314c2895c/docs/index.html" target="_blank">2c39fd1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/9820e7ecbdf8a9ee5bb43821a264239dc9cb5884/docs/index.html" target="_blank">9820e7e</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/abd1ae9a51aab1c5db7dc67ca5e346a76157dbb1/analysis/index.Rmd" target="_blank">abd1ae9</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-11
</td>
<td>
add geno proc info
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/5d71c2e297696ec0a1fd3b33e3407a9b82a9b88d/docs/index.html" target="_blank">5d71c2e</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/c5fe1c231671fddcc4b2855331dd8b33aecd7f4a/analysis/index.Rmd" target="_blank">c5fe1c2</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-10
</td>
<td>
add motif disruption
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/f81a440313fb5dcea5f9e217d50f06ae74d8293e/docs/index.html" target="_blank">f81a440</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/02d97db5936a63f4a465ce7626100689160711a2/analysis/index.Rmd" target="_blank">02d97db</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-07
</td>
<td>
add intron ratio file
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/458e4942294e3b27ad25b0c9d82569afdcf9c5e2/docs/index.html" target="_blank">458e494</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/4c50025c7df6cd909fba5a32d18b7f373f2d7c9f/docs/index.html" target="_blank">4c50025</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/8710e5c8f3701eec64c89bee178bd2840575745c/analysis/index.Rmd" target="_blank">8710e5c</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-06
</td>
<td>
add new analysis for unex v ex
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/0e20f3a630bccbe8a34b1ea813786205bc7bb241/docs/index.html" target="_blank">0e20f3a</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/c7435023b595bbf5d04b24dd229e7ff9fb859653/analysis/index.Rmd" target="_blank">c743502</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-05
</td>
<td>
first pass overlap
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/d30d453e6194b60f5807c553f5ffadb262975022/docs/index.html" target="_blank">d30d453</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/132198920249420deff82fcf50d04d9f3773703c/analysis/index.Rmd" target="_blank">1321989</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-30
</td>
<td>
add chromhmm analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/a71262e12c60d0e4e694df9893e8706035462cff/docs/index.html" target="_blank">a71262e</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/6432c1bbdbcba1645a26f9c236c5bd6aae0b795a/analysis/index.Rmd" target="_blank">6432c1b</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-29
</td>
<td>
inititate nascent analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/28c8ca30c68cc29beccc34cd50d64c0d9d49923e/docs/index.html" target="_blank">28c8ca3</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/a88eedf9feeaeba2af44f429f931361c419ff930/docs/index.html" target="_blank">a88eedf</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/8f883d80c16388df62dd5e0a353736abe5bcb925/analysis/index.Rmd" target="_blank">8f883d8</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-20
</td>
<td>
add overlap analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/d89772dae76346f78daaef40d61bd313034c4966/docs/index.html" target="_blank">d89772d</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/81a3e1646c908c34822af2ce80eac2e07ee602d6/docs/index.html" target="_blank">81a3e16</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/f484dcdb4493c72bb05e76cd64d14389de036457/analysis/index.Rmd" target="_blank">f484dcd</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-15
</td>
<td>
add nascent transcription plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/35b1f6e7a8632c29546f5c5b768a1a3977bb2b74/docs/index.html" target="_blank">35b1f6e</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/90e0c4d3381dc8c977bb44eccfb1766f9811ac63/analysis/index.Rmd" target="_blank">90e0c4d</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-15
</td>
<td>
add reg heatmap
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/bf3a1e0ba9cbfc39facbabc357f4e690c7a05dda/docs/index.html" target="_blank">bf3a1e0</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/77ca26ac20a8a70f3ad011f8dbe1436c3568110b/analysis/index.Rmd" target="_blank">77ca26a</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-14
</td>
<td>
results by logef
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/144c00b0db094769f86c5e6dbb0de49101c87b57/docs/index.html" target="_blank">144c00b</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5e39f1c9463c9ed6b3179631c5876a9b560c08ce/analysis/index.Rmd" target="_blank">5e39f1c</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-08
</td>
<td>
choose pcs and start qtl rerun
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/f5af9c6e7f82c99ee20a28c0d63c1ba8a34275e0/docs/index.html" target="_blank">f5af9c6</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/1ba7d2bed37a985dfd3e436359be43912ef786b3/analysis/index.Rmd" target="_blank">1ba7d2b</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-08
</td>
<td>
add pca
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/9062ebd3af9ca41b40449bb7c915aa3157a1bd27/docs/index.html" target="_blank">9062ebd</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/0e5af5545ff2b1536fe8a3c02bbbe42c33a15a83/analysis/index.Rmd" target="_blank">0e5af55</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-03
</td>
<td>
add utr coverage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/53eeea4cef34210e7926ae287aecef8c382e7a48/docs/index.html" target="_blank">53eeea4</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/2067946979430b43c67b42a864f2976619aa69f4/analysis/index.Rmd" target="_blank">2067946</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-02
</td>
<td>
add old vs new data usage analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/60093ce5e545fc47f274a3f3f76d2bc0c8c93cc1/docs/index.html" target="_blank">60093ce</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/24c2cebc7e4b061243ebb5b04abce358a1035334/analysis/index.Rmd" target="_blank">24c2ceb</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-02
</td>
<td>
add diff iso
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/ac656aad301db960e1e2b218a7235b295395057d/docs/index.html" target="_blank">ac656aa</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/f9b8195e7d9465dff3beb1cb72b790d740f2e1d0/analysis/index.Rmd" target="_blank">f9b8195</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-30
</td>
<td>
understand usage of new pas
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/e3bdc3ada15495b74f56c2b12038d4364c179354/docs/index.html" target="_blank">e3bdc3a</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/be227e79285c0c84178c9136fac1d287a5f40eb7/docs/index.html" target="_blank">be227e7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/b2525c343bc47d98f6ff2e82c969a014866b1c95/docs/index.html" target="_blank">b2525c3</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/9bbe437468e7c6b585b8e832f6cec7ec20a09c93/analysis/index.Rmd" target="_blank">9bbe437</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-24
</td>
<td>
add corr analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/214c05cf4812c042a6351f83b9e41c5c82a62102/docs/index.html" target="_blank">214c05c</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/27b11e33aff06ac75bf10cfbc99b02be56ced69d/analysis/index.Rmd" target="_blank">27b11e3</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-23
</td>
<td>
start signal site analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/1fb70868f617a09060d6e68818511172255910df/docs/index.html" target="_blank">1fb7086</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/e650e08719a9e5e69ce9b3fba86c0f6218fe9d32/docs/index.html" target="_blank">e650e08</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/851c963223c1de49d3f362ddacf6a8877b1087cf/analysis/index.Rmd" target="_blank">851c963</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-22
</td>
<td>
add reads against feature
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/28bd04625d949dd9ab041b7731182dbfaf0d4b4f/docs/index.html" target="_blank">28bd046</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/017f5c0697a9aaddf6f3ae014d8c5e51c9888906/analysis/index.Rmd" target="_blank">017f5c0</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-18
</td>
<td>
add map apa qtl pipeline
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/874081de6302ee3b10d011d4172466258ef2f2a4/docs/index.html" target="_blank">874081d</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/22bfb669998c6ed8e7580d6ca0bab6444d96489a/analysis/index.Rmd" target="_blank">22bfb66</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-18
</td>
<td>
add pas usage qc analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/aab1fcabfe9fa9cdc95d55ac98047034ca5e260d/docs/index.html" target="_blank">aab1fca</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/057dc1a7d10d728fbf408543b730413b962c805b/analysis/index.Rmd" target="_blank">057dc1a</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-17
</td>
<td>
add bam 2 pas analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/fd1f8cb6247fb795ffe19f71517f8ab22708323e/docs/index.html" target="_blank">fd1f8cb</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/f5e23f3477a0ad01175f1189b7638e2b3e9698b1/analysis/index.Rmd" target="_blank">f5e23f3</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-11
</td>
<td>
add fastq 2 bam
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/444f1c71d3fb84f5ba09c2860175bb340ee06174/docs/index.html" target="_blank">444f1c7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/1c955c56c5d95a52e5a91856631d9c34f8c574c5/docs/index.html" target="_blank">1c955c5</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/0f343f83e954f49102656d49fc3196bc577c4a64/analysis/index.Rmd" target="_blank">0f343f8</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-11
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Welcome to my research website.</p>
<div id="preprocessing" class="section level2">
<h2>Preprocessing:</h2>
<p>In the preprocessing steps of the analysis I will go from the fastq files that come off the sequencer to the PAS I will use for the rest of the analysis.</p>
<ul>
<li><p><a href="fastq2bam.html">Process fastq to cleaned bam files</a></p></li>
<li><p><a href="bam2pas.html">Process bam to PAS</a></p></li>
</ul>
</div>
<div id="qc-and-base-visualization" class="section level2">
<h2>QC and base visualization:</h2>
<p>These files create some of the QC metrics I used for the analysis.</p>
<ul>
<li><p><a href="PASusageQC.html">Look at PAS usage differences by gene coverage</a></p></li>
<li><p><a href="Readdistagainstfeatures.html">Look at read distribution accross features</a></p></li>
<li><p><a href="PASdescriptiveplots.html">Make descriptive plots for called PAS</a></p></li>
<li><p><a href="signalsiteanalysis.html">Plot signal site dristributions</a></p></li>
<li><p><a href="corrbetweenind.html">Count and usage correlation</a></p></li>
<li><p><a href="intonRNAratio.html">Intron RNA seq ratios</a></p></li>
<li><p><a href="geno_proc.html">Prepare genotypes for downstream analysis</a></p></li>
<li><p><a href="compareAnnotatedpas.html">Compare to Annotations</a></p></li>
<li><p><a href="UtrPASenrichment.html">3’ UTR PAS enrichment</a></p></li>
<li><p><a href="IntronicPASenrichment.html">Intronic PAS enrichment</a></p></li>
</ul>
</div>
<div id="apaqtl" class="section level2">
<h2>apaQTL</h2>
<ul>
<li><p><a href="mapapaQTL.html">Call apaQTLs in both fractions</a></p></li>
<li><p><a href="choosePCs.html">Look at how many PCs I should include</a></p></li>
<li><p><a href="rerunQTL_changePC.html">Rerun qtl with PCs from previous analysis</a></p></li>
<li><p><a href="QTLlocation.html">Location of QTLs with respect to PAS and gene body</a></p></li>
<li><p><a href="apaQTLoverlap.html">apaQTL in opposite fraction</a></p></li>
<li><p><a href="nonNormQTL.html">Usage QTL analysis to look at non normal effect sizes</a></p></li>
<li><p><a href="QTL_inclusivePASset.html">Specificity of QTL inclusive</a></p></li>
<li><p><a href="snpinSS.html">Are snps in Signal sites likely to be QTLs?</a></p></li>
<li><p><a href="twoRegMech.html">2 Regulatory mechanisms for apaQTLs</a></p></li>
<li><p><a href="PrematureTermQTL.html">Pre-termination QTLs</a></p></li>
<li><p><a href="pttgeneinAPA.html">Overlap pttQTL and apaQTLs</a></p></li>
</ul>
</div>
<div id="differential-isoform-analysis" class="section level2">
<h2>Differential Isoform Analysis</h2>
<ul>
<li><p><a href="DiffIsoAnalysis.html">Run leafcutter to compare fractions</a></p></li>
<li><p><a href="diffiso_logef5.html">Results when logef &gt; 0.5</a></p></li>
<li><p><a href="diffuse_logef1.html">Results when logef &gt; 1</a></p></li>
<li><p><a href="rna_netseq_h3k12ac.html">Regulatory correlation heatmap</a></p></li>
<li><p><a href="nascenttranscription.html">Relationship between APA fraction and nascent transcription</a></p></li>
</ul>
</div>
<div id="characterize-apaqtls-and-compare-to-eqtls" class="section level2">
<h2>Characterize apaQTLs and compare to eQTLs</h2>
<ul>
<li><p><a href="chromHHMQTL.html">Analysis to look at differences in QTL chromHMM categories</a></p></li>
<li><p><a href="exvunexpeQTL.html">Comparing explained and unexplained eGenes</a></p></li>
<li><p><a href="propeQTLs_explained.html">Proportion explained</a></p></li>
<li><p><a href="NuclearSpecAPAqtl.html">Nuclear specific apaQTLs</a></p></li>
<li><p><a href="usageNucspecapaQTLs.html">Nuclear specific apaQTLs and differential usage</a></p></li>
<li><p><a href="NuclearSpecIncludeNotTested.html">New way to look at nuclear specific, include not tested in total</a></p></li>
<li><p><a href="pQTLandeQTLoverlap.html">Comparing protein specific genes with apaQTL</a></p></li>
<li><p><a href="molQTL.html">Rerun eQTLs and pQTLs for use in overlap analysis</a></p></li>
<li><p><a href="eQTLeffectNucspecapa.html">eQTL and pQTL effect sizes for apaQTLs</a></p></li>
<li><p><a href="nucSpecinEQTLs.html">Nuclear specific apaQTL genes in eQTL results</a></p></li>
<li><p><a href="pQTLexampleplot.html">pQTL example</a></p></li>
<li><p><a href="pttQTLinE.html">pttQTL in expression data</a></p></li>
<li><p><a href="gwasOverlap.html">GWAS overlap</a></p></li>
</ul>
</div>
<div id="graph-from-mayher" class="section level2">
<h2>Graph from Mayher</h2>
<ul>
<li><p><a href="PAS_graphs.html">Location and usage of PAS Nuclear Fraction</a></p></li>
<li><p><a href="PAS_graphs_total.html">Location and usage of PAS Total Fraction</a></p></li>
<li><p><a href="reads_graphs.html">Read metadata plots</a></p></li>
</ul>
</div>
<div id="additional-qc-about-mp" class="section level2">
<h2>Additional QC about MP</h2>
<ul>
<li><p><a href="flagMPpas.html">Test number of PAS affected at different filters</a></p></li>
<li><p><a href="version15bpfilter.html">Test if results are robust to cuttoff</a></p></li>
</ul>
</div>
<div id="exploratory-analysis-not-used-in-ms" class="section level2">
<h2>Exploratory analysis not used in MS</h2>
<ul>
<li><p><a href="seqmetadatanalysis.html">Sequencing meta data by batch</a></p></li>
<li><p><a href="NewvOldPeakQC.html">Understand new PAS</a></p></li>
<li><p><a href="recomputeusageold.html">Compare new and old data</a></p></li>
<li><p><a href="reads2UTR.html">Look at coverage in 3’ UTR</a></p></li>
<li><p><a href="additionalCoveragePlots.html">Addition coverage plots</a></p></li>
<li><p><a href="newQTLheatmap.html">Check for batch effects in new QTLs</a></p></li>
<li><p><a href="QTLexampleplots.html">Code to make example plots</a></p></li>
<li><p><a href="motifDisruption.html">Motif distruption as apaQTL mechanism</a></p></li>
<li><p><a href="TF_motifdisrupt.html">Transcription factor motif disruption for QTLs</a></p></li>
<li><p><a href="GO_apaQTLS.html">Gene Ontology</a></p></li>
<li><p><a href="nucintronicanalysis.html">Location of nuclear intronic peaks along introns</a></p></li>
<li><p><a href="verifyFirstintronresult.html">Explore first intron PAS further</a></p></li>
<li><p><a href="nascentRNA.html">Start looking at the nascent seq</a></p></li>
<li><p><a href="HighCrediblePAS.html">Credible set with RNA and signal sites</a></p></li>
</ul>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
