---
title: "ChromHMM enrichment for apaQTLs"
author: "Briana Mittleman"
date: "5/30/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
library(workflowr)
library(reshape2)

```

The goal of this analysis is to see if there is a difference in the QTL chrom hmm locations between the total and nuclear set. I will resample with replacement the number of times for then number of QTLs I have. Then I will count how many QTLs are in each chromHMM region. I will plot these values and include error bars.  

I want one script that will do the replacement, mapping to chromHMM, and append to a final dataframe. This dataframe will have a row for each category and a column for each sample. I can do this using a dictionary with the category as the keys and a list of values representing the number of snps in each catgory.  

I will use pybedtools to overlap the snps with the ChromHMM categories.  

The categories are in: /project2/gilad/briana/genome_anotation_data/GM12878.chromHMM.sort.bed


First I need to make bedfiles for each of the QTL lists.


```{bash,eval=F}
python QTL2bed.py Total

python QTL2bed.py Nuclear 


```



```{bash,eval=F}
mkdir ../data/HMMqtls
sbatch runHMMpermuteAPAqtls.sh
```


```{r}
chromHmm=read.table("../data/HMMqtls/chromHMM_regions.txt", col.names = c("HMMcat", "HMMname"), stringsAsFactors = F)
nucQTLs=read.table("../data/apaQTLs/Nuclear_apaQTLs4pc_5fdr.bed", header = T)
totQTLs=read.table("../data/apaQTLs/Total_apaQTLs4pc_5fdr.bed", header = T)
```

```{r}
TotalRes=read.table("../data/HMMqtls/TotalAPAqtls.HMM1000times.txt", stringsAsFactors = F, col.names =c("HMMcat", seq(1,1000)))

TotalResM=melt(TotalRes, id.vars="HMMcat") %>%inner_join(chromHmm, by="HMMcat") %>%  group_by(HMMname) %>% summarise(Mean=mean(value), SD=sd(value)) %>% mutate(Category="Total_APA")

TotalResMProp=melt(TotalRes, id.vars="HMMcat") %>%inner_join(chromHmm, by="HMMcat") %>%  mutate(nQTL= nrow(totQTLs), propQTL=value/nQTL)%>%  group_by(HMMname) %>% summarise(Mean=mean(propQTL), SD=sd(propQTL)) %>% mutate(Category="Total_APA")
```


```{r}
ggplot(TotalResM, aes(x=HMMname, y=Mean, fill=Category)) + geom_bar(stat="identity")+ geom_errorbar(aes(ymin=Mean-SD, ymax=Mean+SD),width=.5) + theme(axis.text.x=element_text(angle=90, hjust=1))

ggplot(TotalResMProp, aes(x=HMMname, y=Mean, fill=Category)) + geom_bar(stat="identity")+ geom_errorbar(aes(ymin=Mean-SD, ymax=Mean+SD),width=.5) + theme(axis.text.x=element_text(angle=90, hjust=1))

```

```{r}
NuclearRes=read.table("../data/HMMqtls/NuclearAPAqtls.HMM1000times.txt", stringsAsFactors = F, col.names =c("HMMcat", seq(1,1000)))

NuclearResM=melt(NuclearRes, id.vars="HMMcat") %>%inner_join(chromHmm, by="HMMcat") %>%  group_by(HMMname) %>% summarise(Mean=mean(value), SD=sd(value)) %>% mutate(Category="Nuclear_APA")

NuclearResMProp=melt(NuclearRes, id.vars="HMMcat") %>%inner_join(chromHmm, by="HMMcat") %>%  mutate(nQTL= nrow(nucQTLs), propQTL=value/nQTL)%>%  group_by(HMMname) %>% summarise(Mean=mean(propQTL), SD=sd(propQTL)) %>% mutate(Category="Nuclear_APA")
```


```{r}
ggplot(NuclearResM, aes(x=HMMname, y=Mean, fill=Category)) + geom_bar(stat="identity")+ geom_errorbar(aes(ymin=Mean-SD, ymax=Mean+SD),width=.5) + theme(axis.text.x=element_text(angle=90, hjust=1))

ggplot(NuclearResMProp, aes(x=HMMname, y=Mean, fill=Category)) + geom_bar(stat="identity")+ geom_errorbar(aes(ymin=Mean-SD, ymax=Mean+SD),width=.5) + theme(axis.text.x=element_text(angle=90, hjust=1))

```
Next run this for eQTLs  as well 




Run for explained and unexplained eQTLs: 

```{r}
explained=read.table("../data/Li_eQTLs/explained_FDR10.SNPs.noChr.txt", col.names = c("SNPchr", "SNPend")) %>% mutate(SNPstart=as.integer(SNPend)-1, gene="NA", score=".", strand="+") %>% dplyr::select(SNPchr, SNPstart, SNPend, gene, score, strand)

write.table(explained, file="../data/Li_eQTLs/explained_FDR10.SNPs.noChr.bed", col.names = T,row.names = F, quote = F, sep="\t")


unexplained=read.table("../data/Li_eQTLs/unexplained_FDR10.SNPs.noChr.txt", col.names = c("SNPchr", "SNPend")) %>% mutate(SNPstart=as.integer(SNPend)-1, gene="NA", score=".", strand="+") %>% dplyr::select(SNPchr, SNPstart, SNPend, gene, score, strand)

write.table(unexplained, file="../data/Li_eQTLs/unexplained_FDR10.SNPs.noChr.bed", col.names = T,row.names = F, quote = F, sep="\t")

```

```{bash,eval=F}
sbatch runHMMpermuteeQTLS.sh
```

Explained Res
```{r}
explainedRes=read.table("../data/HMMqtls/explainedQTLs.HMM1000times.txt", stringsAsFactors = F, col.names =c("HMMcat", seq(1,1000)))


explainedResM=melt(explainedRes, id.vars="HMMcat") %>% inner_join(chromHmm, by="HMMcat") %>%  group_by(HMMname) %>% summarise(Mean=mean(value), SD=sd(value))  %>% mutate(Category="Explained_eQTL")

explainedResMProp=melt(explainedRes, id.vars="HMMcat") %>%inner_join(chromHmm, by="HMMcat") %>%  mutate(nQTL= nrow(explained), propQTL=value/nQTL)%>%  group_by(HMMname) %>% summarise(Mean=mean(propQTL), SD=sd(propQTL)) %>% mutate(Category="Explained_eQTL")
```

```{r}
ggplot(explainedResM, aes(x=HMMname, y=Mean, fill=Category)) + geom_bar(stat="identity")+ geom_errorbar(aes(ymin=Mean-SD, ymax=Mean+SD),width=.5) + theme(axis.text.x=element_text(angle=90, hjust=1))
```


Unexplained Res

```{r}
unexplainedRes=read.table("../data/HMMqtls/unexplainedeQTLs.HMM1000times.txt", stringsAsFactors = F, col.names =c("HMMcat", seq(1,1000)))

unexplainedResM=melt(unexplainedRes, id.vars="HMMcat") %>%inner_join(chromHmm, by="HMMcat") %>%  group_by(HMMname) %>% summarise(Mean=mean(value), SD=sd(value)) %>% mutate(Category="Unexplained_eQTL")

unexplainedResMProp=melt(unexplainedRes, id.vars="HMMcat") %>%inner_join(chromHmm, by="HMMcat") %>%  mutate(nQTL= nrow(unexplained), propQTL=value/nQTL)%>%  group_by(HMMname) %>% summarise(Mean=mean(propQTL), SD=sd(propQTL)) %>% mutate(Category="Unexplained_eQTL")
```

```{r}
ggplot(unexplainedResM, aes(x=HMMname, y=Mean, fill=Category)) + geom_bar(stat="identity")+ geom_errorbar(aes(ymin=Mean-SD, ymax=Mean+SD),width=.5) + theme(axis.text.x=element_text(angle=90, hjust=1))
```


```{r}
AllQTLs=as.data.frame(rbind(NuclearResM,TotalResM,explainedResM,unexplainedResM)) 

AllQTLsProp=as.data.frame(rbind(NuclearResMProp,TotalResMProp,explainedResMProp,unexplainedResMProp)) 
```

```{r}
ggplot(AllQTLs, aes(x=HMMname, y=Mean, fill=Category)) + geom_bar(stat="identity", position="dodge")+ geom_errorbar(aes(ymin=Mean-SD, ymax=Mean+SD),width=.5, position = position_dodge(0.75)) + theme(axis.text.x=element_text(angle=90, hjust=1))+labs(y="Number of QTLs", x="ChromHMM Category", title="ChromHMM locations for all QTLs")
```

```{r}
ggplot(AllQTLsProp, aes(x=HMMname, y=Mean, fill=Category)) + geom_bar(stat="identity", position="dodge")+ geom_errorbar(aes(ymin=Mean-SD, ymax=Mean+SD),width=.5, position = position_dodge(0.75)) + theme(axis.text.x=element_text(angle=90, hjust=1))+labs(y="Proportion of QTLs", x="ChromHMM Category", title="ChromHMM locations for all QTLs")
```
Plot the 95% confidence interval:  


```{r}
unexplainedResMPropQuant=melt(unexplainedRes, id.vars="HMMcat") %>%inner_join(chromHmm, by="HMMcat") %>%  mutate(nQTL= nrow(unexplained), propQTL=value/nQTL)%>%  group_by(HMMname) %>% summarise(list(enframe(quantile(propQTL, probs=c(0.25,0.5,0.75))))) %>% unnest() %>% spread(name, value)%>% mutate(category="UnexplainedeQTL")
colnames(unexplainedResMPropQuant)=c("HMMname", "twentyfive", "fifty", "seventyfive", "category")

explainedResMPropQuant=melt(explainedRes, id.vars="HMMcat") %>%inner_join(chromHmm, by="HMMcat") %>%  mutate(nQTL= nrow(explained), propQTL=value/nQTL)%>%  group_by(HMMname) %>% summarise(list(enframe(quantile(propQTL, probs=c(0.25,0.5,0.75))))) %>% unnest() %>% spread(name, value)%>% mutate(category="ExplainedeQTL")
colnames(explainedResMPropQuant)=c("HMMname", "twentyfive", "fifty", "seventyfive", "category")


NuclearResMPropQuant=melt(NuclearRes, id.vars="HMMcat") %>%inner_join(chromHmm, by="HMMcat") %>%  mutate(nQTL= nrow(nucQTLs), propQTL=value/nQTL)%>%  group_by(HMMname) %>% summarise(list(enframe(quantile(propQTL, probs=c(0.25,0.5,0.75))))) %>% unnest() %>% spread(name, value)%>% mutate(category="NuclearApa")
colnames(NuclearResMPropQuant)=c("HMMname", "twentyfive", "fifty", "seventyfive", "category")


TotalResMPropQuant=melt(TotalRes, id.vars="HMMcat") %>%inner_join(chromHmm, by="HMMcat") %>%  mutate(nQTL= nrow(totQTLs), propQTL=value/nQTL)%>%  group_by(HMMname) %>% summarise(list(enframe(quantile(propQTL, probs=c(0.25,0.5,0.75))))) %>% unnest() %>% spread(name, value) %>% mutate(category="TotalApa")
colnames(TotalResMPropQuant)=c("HMMname", "twentyfive", "fifty", "seventyfive", "category")
    
                                                                               
```

```{r}

AllQTLsProp95=as.data.frame(rbind(NuclearResMPropQuant,unexplainedResMPropQuant,explainedResMPropQuant)) 
AllQTLsProp95$category=factor(AllQTLsProp95$category, levels=c("NuclearApa", "UnexplainedeQTL","ExplainedeQTL"))

```


```{r}
ggplot(AllQTLsProp95, aes(x=HMMname, y=fifty, fill=category)) + geom_bar(stat="identity", position="dodge")+ geom_errorbar(aes(ymin=twentyfive, ymax=seventyfive),width=.5, position = position_dodge(0.9)) + theme(axis.text.x=element_text(angle=45, hjust=1),legend.position = "top", legend.title = element_blank())+labs(y="Proportion of QTLs", x="ChromHMM Category", title="ChromHMM locations for all QTLs") + scale_fill_brewer(palette="Dark2")
```

```{r}
AllQTLsProp95_certcat=AllQTLsProp95 %>% filter(HMMname %in% c("Active_Promoter", "Strong_Enhancer","Txn_Elongation", "Weak_Enhancer", "Weak_Promter","Poised_Promoter", "Weak_Txn" ))

AllQTLsProp95_certcat$HMMname= factor(AllQTLsProp95_certcat$HMMname, levels=c('Strong_Enhancer','Weak_Enhancer','Poised_Promoter',"Active_Promoter", 'Weak_Promter','Txn_Elongation',  'Weak_Txn' ))


chromhmmplot=ggplot(AllQTLsProp95_certcat, aes(x=HMMname, y=fifty, fill=category)) + geom_bar(stat="identity", position="dodge")+ geom_errorbar(aes(ymin=twentyfive, ymax=seventyfive),width=.5, position = position_dodge(0.9)) + theme(axis.text.x=element_text(angle=45, hjust=1),legend.position = "top", legend.title = element_blank())+labs(y="Proportion of QTLs", x="ChromHMM Category", title="ChromHMM locations for all QTLs") + scale_fill_brewer(palette="Dark2",labels=c("apaQTL","Unexaplined eQTL", "Explained eQTL"))+scale_x_discrete(labels= c("Strong Enhancer", "Weak Enhancer", "Poised Promoter", "Active Promoter", "Transcription Elongation", "Weak Transcription"))

chromhmmplot


```



```{r figure3D, include=FALSE, dev="pdf", fig.height=5, fig.width=5, crop=FALSE}
chromhmmplot
```



Plot only the total and nuclear:  

```{r}
APAqtlsProp95=as.data.frame(rbind(NuclearResMPropQuant,TotalResMPropQuant)) 

ggplot(APAqtlsProp95, aes(x=HMMname, y=fifty, fill=category)) + geom_bar(stat="identity", position="dodge")+ geom_errorbar(aes(ymin=twentyfive, ymax=seventyfive),width=.5, position = position_dodge(0.9)) + theme(axis.text.x=element_text(angle=45, hjust=1),legend.position = "top", legend.title = element_blank())+labs(y="Proportion of QTLs", x="ChromHMM Category", title="ChromHMM locations for apaQTLs") +scale_fill_manual(values=c("deepskyblue3","darkviolet"))
```

