---
title: "Colocalization"
author: "Briana Mittleman"
date: "2/19/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Attempt a colocalziation analysis for the eQTLs and the snps in the nuclear QTL data.  

```{bash,eval=F}
mkdir ../data/coloc
wget http://eqtl.uchicago.edu/jointLCL/output_RNAseqGeuvadis_PC14.txt
```

```{r}
#install.packages("coloc")
library(snpStats)
library(cowplot)
library(data.table)
library(tidyverse)
library("coloc")
```
First change the names of the rsid and gene. 
```{r}

geneNames=read.table("../../genome_anotation_data/ensemble_to_genename.txt", sep="\t", col.names = c('gene_id', 'GeneName', 'source' ),stringsAsFactors = F, header = T)
RSID=read.table("/project2/gilad/briana/li_genotypes/RSID2snploc.txt",header = T, stringsAsFactors = F)
#ed 's/^chr//' output_RNAseqGeuvadis_PC14.txt  > output_RNAseqGeuvadis_PC14.noChr.txt 

eQTLres=read.table("../data/coloc/output_RNAseqGeuvadis_PC14.noChr.txt",header = T, stringsAsFactors = F)  %>% separate(snps,into=c("chr", "pos"), sep="\\.")%>% separate(gene,into=c("gene_id", "extra"), sep="\\.")  %>% mutate(snp=paste(chr,pos, sep=":")) %>% inner_join(RSID,by="snp") %>% inner_join(geneNames, by="gene_id") %>% select(RSID, GeneName,pvalue, beta)

```
I will be able to seperate these for the coloc analysis. I need the PAS gene pairs.  

```{r}
PASgene=read.table("../data/eQTL_LZ/PasGENEsnpstoUse.txt",stringsAsFactors = F, col.names = c("GeneName", "PAS", "RSID") ) 
```

Test if these are in the file above:  

```{r}
eQTLres_genesnp=eQTLres %>% mutate(test=paste(GeneName, RSID, sep="_")) %>% select(test)
PASgene_test= PASgene %>%  mutate(test=paste(GeneName, RSID, sep="_")) %>% select(test)


PASgene_indata= PASgene_test %>% inner_join(eQTLres_genesnp, by="test")

PASgene_indata= PASgene_test %>% anti_join(eQTLres_genesnp, by="test")

```

181 of the snp gene pairs are in this data.  

Test on one first.  c10orf88 


Need: 

p values for each SNP
each SNPâ€™s minor allele frequency
sample size

C10orf88_rs7904973

```{r}
PASgene_c10orf88= PASgene %>% filter(GeneName=="C10orf88")
PASgene_c10orf88
```

```{r}
PeakData=read.table("../data/eQTL_LZ/NuclearAssoc/peak19682_NuclearResults4LZ.txt",col.names = c("RSID","Pvalue_APA"),stringsAsFactors = F) %>% mutate(sampleSizeAPA=52)

GeneData=eQTLres %>% filter(GeneName=="C10orf88")
```

Number of overlapping snps:  

```{r}
length(intersect(PeakData$RSID,GeneData$RSID ))

nrow(GeneData)
nrow(PeakData)
```

```{r}
ind=read.table("../data/molPhenos/RNAhead.txt", header = F)
dim(ind)[2]-4
```

```{r}
Genenomnames=c("gene", 'snp','dist', 'pval', 'slope')
c10orf88eqtl=read.table("../data/coloc/eQTL_c10orf88.txt",col.names = Genenomnames,stringsAsFactors = F) %>% mutate(SampleSizeE=69) %>% inner_join(RSID, by="snp")


```
Get freq for the snps in allChrom.dose.filt.vcf. these are the filtered snps from john. not the li snps but should include those i need and they have the same 119 ind. 

```{bash,eval=F}
module load vcftools


vcftools --vcf allChrom.dose.filt.vcf   --freq --out allChrom.dose.filt_freq

```






```{r}
snpmaf=fread("/project2/gilad/briana/YRI_geno_hg19/allChrom.dose.filt_freq.frq",skip = 1, col.names = c("chr", "loc", "nAlles", "nchr", "allele1", "allele2")) %>% mutate(snp=paste(chr, loc, sep=":")) %>% separate("allele2", into=c("allele", "MAF2"), sep=":") %>%separate("allele1", into=c("allele", "MAF1"), sep=":")  %>% mutate(MAF= ifelse(MAF1 < MAF2, MAF1, MAF2))%>%  inner_join(RSID, by="snp") %>% select(RSID, MAF)
```
```{r}

All= PeakData %>% inner_join(snpmaf,by="RSID") %>% inner_join(c10orf88eqtl, by="RSID")
All$Pvalue=as.numeric(as.character(All$Pvalue))
All$MAF=as.numeric(as.character(All$MAF))

```

Join with eQTL and apa  

```{r}
my.res <- coloc.abf(dataset1=list(pvalues=All$Pvalue,N=52,type="quant"),
                    dataset2=list(pvalues=All$pval,N=69,type="quant"),
                    MAF=All$MAF)

C10orf88= as.data.frame(t(my.res$summary ))%>% mutate(gene="C10orf88")
C10orf88
```


This isnt a great result but it works.   


I need to do this for all of them. 


Write MAF data:  

```{r}
write.table(snpmaf, "../data/coloc/snpmaf.txt", col.names = T, row.names = F, quote = F)
```

```{bash,eval=F}
sbatch ColocApAeQTL.sh
cat ../data/coloc/*colocRes.txt > ../data/coloc/AllAssoccolocRes.txt
```

H0: neither trait has a genetic association in the region
H1: only trait 1 has a genetic association in the region
H2: only trait 2 has a genetic association in the region
H3: both traits are associated, but with different causal variants
H4: both traits are associated and share a single causal variant


I think you should use PP4/(PP4+PP3)

and count only when PP4+PP3 > 0.2

```{r}
ColocRES=read.table("../data/coloc/AllAssoccolocRes.txt", col.names = c("nsnp",'Neither', 'apaQTL', 'eQTL', 'DifferentSnps', 'Coloc','gene'),stringsAsFactors = F) %>% select(-nsnp) %>% mutate(three_four=DifferentSnps+Coloc, Prop=Coloc/three_four)

ColocRES_filt=ColocRES %>% filter(three_four>.2)
ColocRES_filt1=ColocRES %>% filter(three_four>.1)
ColocRES_filt3=ColocRES %>% filter(three_four>.3)
ColocRES_filt4=ColocRES %>% filter(three_four>.4)

```

```{r}
all=ggplot(ColocRES,aes(x=Prop)) +geom_histogram(bins=50) + labs(y="Number of Genes", x="PP4/(PP3+PP4)", title=paste("All Genes ", nrow(ColocRES), sep=": "))

one=ggplot(ColocRES_filt1,aes(x=Prop)) +geom_histogram(bins=50)+ labs(y="Number of Genes", x="PP4/(PP3+PP4)", title=paste("PP3+PP4 > .1 : ", nrow(ColocRES_filt1), "genes" , sep=" "))


two=ggplot(ColocRES_filt,aes(x=Prop)) +geom_histogram(bins=50)+ labs(y="Number of Genes", x="PP4/(PP3+PP4)", title=paste("PP3+PP4 > .2 :", nrow(ColocRES_filt), "genes", sep=" "))


three=ggplot(ColocRES_filt3,aes(x=Prop)) +geom_histogram(bins=50)+ labs(y="Number of Genes", x="PP4/(PP3+PP4)", title=paste("PP3+PP4 > .3 : ", nrow(ColocRES_filt3), "genes", sep=" "))

four=ggplot(ColocRES_filt4,aes(x=Prop)) +geom_histogram(bins=50)+ labs(y="Number of Genes", x="PP4/(PP3+PP4)", title=paste("PP3+PP4 > .4 :", nrow(ColocRES_filt4), 'genes', sep=" "))
```


```{r}
plot_grid(all, one, two, three, four)
```

Try with a higher powered eQTL set.  


```{bash,eval=F}
mkdir ../data/GeuvadiseQTL/
cp  /project2/yangili1/zpmu/geuvadis/results/eqtl_new/YRI/nom_YRI_9phenoPC_*.txt.gz ../data/GeuvadiseQTL/
gunzip ../data/GeuvadiseQTL/*
cat ../data/GeuvadiseQTL/nom_YRI* > ../data/GeuvadiseQTL/nom_YRI_9phenoPCALL.txt

```

MAF for these data:  

```{bash,eval=F}
cp  /project2/yangili1/zpmu/geuvadis/data/new_fastqtl/chr*.dose.r2.nodup.filtered.recode.YRI.frq.gz ../data/GeuvadiseQTL/
  
gunzip ../data/GeuvadiseQTL/*.gz

cat ../data/GeuvadiseQTL/*ose.r2.nodup.filtered.recode.YRI.frq > ../data/GeuvadiseQTL/allMAFRes.txt

mkdir ../data/coloc_PM

sbatch ColocApAeQTL_PM.sh

cat ../data/coloc_PM/*colocRes.txt > ../data/coloc_PM/AllAssoccolocResPM.txt
```

```{r}
ColocRESPM=read.table("../data/coloc_PM/AllAssoccolocResPM.txt", col.names = c("nsnp",'Neither', 'apaQTL', 'eQTL', 'DifferentSnps', 'Coloc','gene'),stringsAsFactors = F) %>% select(-nsnp) %>% mutate(three_four=DifferentSnps+Coloc, Prop=Coloc/three_four)

ColocRESPM_filt=ColocRESPM %>% filter(three_four>.2)
ColocRESPM_filt1=ColocRESPM %>% filter(three_four>.1)
ColocRESPM_filt3=ColocRESPM %>% filter(three_four>.3)
ColocRESPM_filt4=ColocRESPM %>% filter(three_four>.4)

```

```{r}
allPM=ggplot(ColocRESPM,aes(x=Prop)) +geom_histogram(bins=50) + labs(y="Number of Genes", x="PP4/(PP3+PP4)", title=paste("All Genes ", nrow(ColocRESPM), sep=": "))

onePM=ggplot(ColocRESPM_filt1,aes(x=Prop)) +geom_histogram(bins=50)+ labs(y="Number of Genes", x="PP4/(PP3+PP4)", title=paste("PP3+PP4 > .1 : ", nrow(ColocRESPM_filt1), "genes" , sep=" "))


twoPM=ggplot(ColocRESPM_filt,aes(x=Prop)) +geom_histogram(bins=50)+ labs(y="Number of Genes", x="PP4/(PP3+PP4)", title=paste("PP3+PP4 > .2 :", nrow(ColocRESPM_filt), "genes", sep=" "))


threePM=ggplot(ColocRESPM_filt3,aes(x=Prop)) +geom_histogram(bins=50)+ labs(y="Number of Genes", x="PP4/(PP3+PP4)", title=paste("PP3+PP4 > .3 : ", nrow(ColocRESPM_filt3), "genes", sep=" "))

fourPM=ggplot(ColocRESPM_filt4,aes(x=Prop)) +geom_histogram(bins=50)+ labs(y="Number of Genes", x="PP4/(PP3+PP4)", title=paste("PP3+PP4 > .4 :", nrow(ColocRESPM_filt4), 'genes', sep=" "))
```

```{r}
plot_grid(allPM, onePM, twoPM, threePM, fourPM)
```


```{r}
ggplot(ColocRESPM_filt,aes(x=Prop)) +geom_histogram(bins=50)+ labs(y="Number of Genes", x="PP4/(PP3+PP4)", title=paste("Colocalization results for apaQTLs and eQTLs \nPP3+PP4 > .2 :", nrow(ColocRESPM_filt), "genes", sep=" "))
```
Which genes are these:  

```{r}
ColocRESPM_filt5= ColocRESPM_filt %>% filter(Prop>.5) 
ColocRESPM_filt5$gene
length(ColocRESPM_filt5$gene)
```
Are these unexplained:  

```{r}
UexPe=read.table("../data/Li_eQTLs/UnexplainedEgenes.txt",col.names = "GeneName") %>% filter(GeneName %in% ColocRESPM_filt5$gene)
UexPe
```

