---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
---

Welcome to my research website.


##Preprocessing:

In the preprocessing steps of the analysis I will go from the fastq files that come off the sequencer to the PAS I will use for the rest of the analysis. 

* [Process fastq to cleaned bam files](fastq2bam.html)

* [Process bam to PAS](bam2pas.html) 


##QC and base visualization:

These files create some of the QC metrics I used for the analysis.

* [Look at PAS usage differences by gene coverage](PASusageQC.html) 

* [Look at read distribution accross features](Readdistagainstfeatures.html) 

* [RNA and 3' seq plots with 5 individuals](newDTplots.html) 

* [Make descriptive plots for called PAS](PASdescriptiveplots.html)  

* [Plot signal site dristributions](signalsiteanalysis.html)  

* [Count and usage correlation](corrbetweenind.html)  

* [Intron RNA seq ratios](intonRNAratio.html)

* [Prepare genotypes for downstream analysis](geno_proc.html)

* [Compare to Annotations](compareAnnotatedpas.html)  

* [3' UTR PAS enrichment](UtrPASenrichment.html)

* [Intronic PAS enrichment](IntronicPASenrichment.html)



##apaQTL  


* [Call apaQTLs in both fractions](mapapaQTL.html)  

* [Look at how many PCs I should include](choosePCs.html)

* [Rerun qtl with PCs from previous analysis](rerunQTL_changePC.html)

* [Location of QTLs with respect to PAS and gene body](QTLlocation.html)

* [apaQTL in opposite fraction](apaQTLoverlap.html)

* [Usage QTL analysis to look at non normal effect sizes](nonNormQTL.html)

* [Specificity of QTL inclusive](QTL_inclusivePASset.html)

* [Are snps in Signal sites likely to be QTLs?](snpinSS.html)

* [2 Regulatory mechanisms for apaQTLs](twoRegMech.html)

* [Pre-termination QTLs](PrematureTermQTL.html)

* [Overlap pttQTL and apaQTLs](pttgeneinAPA.html)

##Differential Isoform Analysis  

* [Run leafcutter to compare fractions](DiffIsoAnalysis.html)  

* [Results when logef > 0.5](diffiso_logef5.html)  

* [Results when logef > 1](diffuse_logef1.html) 

* [Regulatory correlation heatmap](rna_netseq_h3k12ac.html)  

* [Relationship between APA fraction and nascent transcription](nascenttranscription.html)  


##Characterize apaQTLs and compare to eQTLs 

* [Analysis to look at differences in QTL chromHMM categories](chromHHMQTL.html)  

* [Comparing explained and unexplained eGenes](exvunexpeQTL.html)  

* [Proportion explained](propeQTLs_explained.html)

* [Nuclear specific apaQTLs](NuclearSpecAPAqtl.html)

* [Nuclear specific apaQTLs and differential usage](usageNucspecapaQTLs.html)

* [New way to look at nuclear specific, include not tested in total](NuclearSpecIncludeNotTested.html)

* [Comparing protein specific genes with apaQTL](pQTLandeQTLoverlap.html)

* [Rerun eQTLs and pQTLs for use in overlap analysis](molQTL.html)

* [eQTL and pQTL effect sizes for apaQTLs](eQTLeffectNucspecapa.html)

* [Nuclear specific apaQTL genes in eQTL results](nucSpecinEQTLs.html)

* [pQTL example](pQTLexampleplot.html)

* [pttQTL in expression data](pttQTLinE.html)

* [GWAS overlap](gwasOverlap.html)


##Graph from Mayher   

* [Location and usage of PAS Nuclear Fraction](PAS_graphs.html)  

* [Location and usage of PAS Total Fraction](PAS_graphs_total.html)  

* [Read metadata plots](reads_graphs.html)

##Additional QC about MP

* [Test number of PAS affected at different filters](flagMPpas.html)

* [Test if results are robust to cuttoff](version15bpfilter.html)

##Exploratory analysis not used in MS  

* [Sequencing meta data by batch](seqmetadatanalysis.html)    

* [Understand new PAS](NewvOldPeakQC.html)  

* [Compare new and old data](recomputeusageold.html)

* [Look at coverage in 3' UTR](reads2UTR.html)  

* [Addition coverage plots](additionalCoveragePlots.html)

* [Check for batch effects in new QTLs](newQTLheatmap.html)  

* [Code to make example plots](QTLexampleplots.html)

* [Motif distruption as apaQTL mechanism](motifDisruption.html)  

* [Transcription factor motif disruption for QTLs](TF_motifdisrupt.html)

* [Gene Ontology](GO_apaQTLS.html)

* [Location of nuclear intronic peaks along introns](nucintronicanalysis.html)  

* [Explore first intron PAS further](verifyFirstintronresult.html)

* [Are PAS enriched in first intron](checkfirstintron.html)

* [Start looking at the nascent seq](nascentRNA.html)

* [Credible set with RNA and signal sites](HighCrediblePAS.html)


Revisions (Thank you to the reviewers for suggestions)  

* [Decay and stability of transcripts](decayAndStability.html)  

* [LD score regression](LDregress.html)  

* [Variance in eQTLs explained by apaQTL](varEqtl2varAPAqtl.html)   

* [Variance in APA and variance in ribosome occupancy](varAPAvarRibo.html)  

* [Conservation of PAS](ConservationPAS.html)  

* [Explore number of PAS and other parameters](ExploreNpas.html)  

* [Tissue specificity for apaQTL PAS](apaQTL_tissueSpec.html)  

* [RNA binding proteins](RNAbinding.html)  

* [TSS analysis](TSS.html)  

* [Base composition around PAS](BaseComposition.html)  

* [apaQTL effect sizes by eQTL set](apabyeQTLstatus.html)

* [miRNA binding site disruption as QTL mechanism](miRNAdisrupt.html)  

* [Compare Effect Sizes for total and nuclear](TotNucQTLeffect.html)  

* [Mechanism for each expression independent apaQTL](mechEindepQTL.html)  

