---
title: "apaQTLs"
author: "Briana Mittleman"
date: "4/18/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(tidyverse)
library(reshape2)
library(workflowr)
library(cowplot)
```

In this analysis I will call apaQTls in both fractions. I will start with the phenotype files and normalized the counts using the leafcutter package in order to run the fastq QTL mapper.  


##Prepare phenotypes for QTL- phenotype dir  
It is best to run this analysis in the data/phenotype_5perc directory. I have copied the leafcutter prepare_phenotype_table.py to the code directroy to use here.  


```{bash,eval=F}
#!/bin/bash
#python2 

gzip APApeak_Phenotype_GeneLocAnno.Total.5perc.fc
gzip APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc

python ../../code/prepare_phenotype_table.py APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz
python ../../code/prepare_phenotype_table.py APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz

```


This will output bash scripts to run. 

```{bash,eval=F}
#three-prime-env

sh APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz_prepare.sh
sh APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz_prepare.sh

```


Subset the PCs to use the first 2 in the qtl calling:  

```{bash,eval=F}
#three-prime-env

head -n 3 APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz.PCs > APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz.2PCs
head -n 3 APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz.PCs > APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz.2PCs

```

##Call QTLs- code dir  

Next I will need to make a sample list. From the code directory: 

```{bash,eval=F}
python makeSampleList.py
```

*remove 19092*
 
 
Prepare directroy 
```{bash,eval=F}
mkdir ../data/apaQTLNominal
mkdir ../data/apaQTLPermuted
```


Run the code to call QTLs within 1mb of each PAS peak. I run both a nominal pass and a permuted pas. The permulted pas chosses the best snp for each peak gene pair. 


```{bash,eval=F}
sbatch apaQTL_Nominal.sh
sbatch apaQTL_permuted.sh
```


Concatinate all of the results in the permuted set. I do this so I can account for multiple testing with the benjamini hochberg test.  

Concatinate results in permuted directory:  
```{bash,eval=F}
cat APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz.qqnorm_chr* > APApeak_Phenotype_GeneLocAnno.Total_permRes.txt

cat APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz.qqnorm_chr* > APApeak_Phenotype_GeneLocAnno.Nuclear_permRes.txt
 
```


Run correction script  
```{bash,eval=F}
Rscript apaQTLCorrectPvalMakeQQ.R  
```


##Evaluation results   

```{r}
totRes=read.table("../data/apaQTLPermuted/APApeak_Phenotype_GeneLocAnno.Total_permResBH.txt", stringsAsFactors = F, header = T) %>% separate(pid, into=c("Chr", "Start", "End", "PeakID"), sep=":") %>% separate(PeakID, into=c("Gene", "Loc", "Strand","Peak"), sep="_")
```


Total Apa QTLs   
```{r}
TotQTLs= totRes %>% filter(-log10(bh)>=1)
nrow(TotQTLs)
```

apaQTL genes:  

```{r}
TotQTLs_gene=TotQTLs %>% group_by(Gene)  %>% summarise(nQTL=n())

summary(TotQTLs_gene$nQTL)

hist(TotQTLs_gene$nQTL)
```


Location distribution for peaks:

```{r}
TotQTLs_loc= TotQTLs %>% group_by(Loc) %>% summarise(nLoc=n()) %>% mutate(PropLoc=nLoc/nrow(TotQTLs))


totQTLloc=ggplot(TotQTLs_loc, aes(x=Loc, y=PropLoc, fill=Loc)) + geom_bar(stat = "Identity") + labs(x="Location of Significant Peak", y="Proportion of QTLs", title="Total QTL peak distribution")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))
```

```{r}
nucRes=read.table("../data/apaQTLPermuted/APApeak_Phenotype_GeneLocAnno.Nuclear_permResBH.txt", stringsAsFactors = F, header = T) %>% separate(pid, into=c("Chr", "Start", "End", "PeakID"), sep=":") %>% separate(PeakID, into=c("Gene", "Loc", "Strand","Peak"), sep="_")
```


Nuclear Apa QTLs   
```{r}
NucQTLs= nucRes %>% filter(-log10(bh)>=1)
nrow(NucQTLs)
```

apaQTL genes:  

```{r}
NucQTLs_gene= NucQTLs %>% group_by(Gene)  %>% summarise(nQTL=n())

summary(NucQTLs_gene$nQTL)

hist(NucQTLs_gene$nQTL)
```

Location distribution for peaks:

```{r}
NucQTLs_loc= NucQTLs %>% group_by(Loc) %>% summarise(nLoc=n()) %>% mutate(PropLoc=nLoc/nrow(NucQTLs))


nucQTLloc=ggplot(NucQTLs_loc, aes(x=Loc, y=PropLoc, fill=Loc)) + geom_bar(stat = "Identity") + labs(x="Location of Significant Peak", y="Proportion of QTLs", title="Nuclear QTL peak distribution")+theme(axis.text.x = element_text(angle = 90, hjust = 1))

```


```{r}
plot_grid(totQTLloc, nucQTLloc)
```

```{r}
write.table(TotQTLs, file="../data/apaQTLs/Total_apaQTLs_5fdr.txt", col.names = T, row.names = F, quote=F)

write.table(NucQTLs, file="../data/apaQTLs/Nuclear_apaQTLs_5fdr.txt", col.names = T, row.names = F, quote=F)

```

