---
title: "Mechanism for each expression independent apaQTL"
author: "Briana Mittleman"
date: "2/3/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this analysis I will pull in the apaQTLs that are pQTLs or riboQTLs but are not eQTLs. I will look at their UTRs to think about mechanisms.  

```{r}
library(workflowr)
library(tidyverse)
```

```{r}
QTLs=read.table("../data/ExpressionIndependentapaQTLs.txt",header = T,stringsAsFactors = F)



QTLs %>% select(Gene, PAS_ID, apa_EffectSize,protein_EffectSize,protein_pvalue,ribo_EffectSize,ribo_pvalue)
```

DIP2B: QTL in proximal UTR, hsa-miR-143-3p, hsa-miR-6088, hsa-miR-4770, miRNAs are upstream of first 

The K box (cTGTGATa) is present in one or more copies in many of these 3'UTRs and mediates negative post-transcriptional regulation, mainly effected by decreased transcript levels.

C12orf73: 2nd of 3 UTR PAS:
- POLR2A binding sites 
-hsa-miR-9-5p 

OAS1:  

IFI44L: between 2 UTR FOS binding site 


STXBP3: PUM1 binding in 3' UTR, translational regulator of specific mRNAs by binding to their 3' untranslated regions, reported kbox, many miRNAs conserved (miR-15-5p/16-5p/195-5p/424-5p/497-5p)  

RBBP5: extended isoform,reported Brd-Box (Brd)  The Brd box (AGCTTTA) is present in one or more copies in many of these 3'UTRs and mediates negative post-transcriptional regulation by affecting transcript stability and translational efficiency. hsa-miR-338-3p

PSMF1: UNR binding site: In humans, UNR has been implicated in the destabilization of c-fos mRNA (Chang et al., 2004) and the activation of translation driven by the IRESs of a number of transcripts, including c-myc, miR-103-3p/107

MGME1- RBP binding for AQR, NIPBL, FUS (no DB results)

FAM207A- (no DB results)

APOL2- reported UNR binding site, SAFB binding between PAS in UTR

MPHOSPH10-

SACM1L-miR-181-5p

EIF2A- annotated Brd-Box (Brd) site between middle 2 utr binding sites, reported kbox as well 


PCYT1A- (no DB results) many binding sites between annotated utr and end PAS, miR-142-3p.2

HTT: BRD-BOX, hsa-miR-216b-5p


STAP1- LTR/ERVL-MaLR repete element 


NUP155-KBOX, BRD BOX , GLUT1  element, miR-199-3p, hsa-miR-3129-5p

ERAP1-MBE element - related to temporal regulation ,hsa-miR-4770, hsa-miR-143-3p, hsa-miR-6088

SQSTM1- FLAM_A - SINE/Alu repeate, GY-BOX- involved in RNA duplexes, hsa-miR-7-5p

BTBD9-  GY-BOX, conserved hsa-miR-212-5p binding site , hsa-miR-212-5p

STX17- GY-BOX, BRD BOX, k-box, miR-15-5p/16-5p/195-5p/424-5p/497-5p

Lit search for BRD box and K box only has results for human, info about regulation in drosophila 



longer isoform, decreased protein: 
* DIP2B  

* IF144L  

* RBBP5  

* STAP1  

* FAM207A  

* APOL2  

* EIF2A  

* STXB  

longer isoform, increased protein:  

* C12orf72

* OAS1  

* PSMF1  

* HTT  

* ERAP 1 

* SQSTM1  

Intronic or coding PAS, decreased protein:  

* KPNB1  

* CCDC47  

* DOCK7  

* MGME1  

* PCYT1A  

* BTBD9  

Intronic or coding PAS, increased protein:  

* NUP155  

* MPHOSPH10


I have the miRNA sites. I will look for the following genes. * DIP2B  

* IF144L, RBBP5, STAP1, FAM207A , APOL2  , EIF2A  , STXB  
```{r}
predMImech=c('IF144L', 'RBBP5', 'STAP1', 'FAM207A' , 'APOL2'  , 'EIF2A'  , 'STXB')

miRNAtargets=read.table("../data/miRNAbinding/Targets_CS_pctiles.hg19.consFam.consSite.bed", stringsAsFactors = F,col.names = c("chr", "start", "end", "ID", "score", "strand", "thikStart", "thinkEnd", "RGB", "blockcount", "blocksize", "blockstart")) %>% separate(ID, into= c("gene", "miRNA"), sep=":") %>% filter(gene %in% predMImech)

miRNAtargets
```


It looks like there are many binding sites for miRNAs in RBBP5. I will look at where these are in relation to the QTL.  

```{r}
miRNAtargets_br=read.table("../data/miRNAbinding/Targets_CS_pctiles.hg19.broadConsFam.consSite.bed", stringsAsFactors = F,col.names = c("chr", "start", "end", "ID", "score", "strand", "thikStart", "thinkEnd", "RGB", "blockcount", "blocksize", "blockstart")) %>% separate(ID, into= c("gene", "miRNA"), sep=":") %>% filter(gene %in% predMImech)

miRNAtargets_br
```

This doesnt look like the explanation here. I am expanding to the unconserved and other families. 


```{r}
miRNAtargets_noconsite=  read.table("../data/miRNAbinding/Targets_CS_pctiles.hg19.consFam.nonConsSite.bed", stringsAsFactors = F,col.names = c("chr", "start", "end", "ID", "score", "strand", "thikStart", "thinkEnd", "RGB", "blockcount", "blocksize", "blockstart")) %>% separate(ID, into= c("gene", "miRNA"), sep=":") %>% filter(gene %in% predMImech)

miRNAtargets_noconsite
```
There are many miRNA sites for conserved family non conserved sites between the UTR isoform and the end isoform for fam207A. This is a good example.  

The non conserved sites are very common in these UTRs. Not sure we should use this as the explanation.  



```{r}
miRNAtargets_nonConfam=  read.table("../data/miRNAbinding/Targets_CS_pctiles.hg19.nonConsFam.consSite.bed", stringsAsFactors = F,col.names = c("chr", "start", "end", "ID", "score", "strand", "thikStart", "thinkEnd", "RGB", "blockcount", "blocksize", "blockstart")) %>% separate(ID, into= c("gene", "miRNA"), sep=":") %>% filter(gene %in% predMImech)

miRNAtargets_nonConfam

miRNAtargets_nonConfamnoconsite=  read.table("../data/miRNAbinding/Targets_CS_pctiles.hg19.nonConsFam.nonConsSite.bed", stringsAsFactors = F,col.names = c("chr", "start", "end", "ID", "score", "strand", "thikStart", "thinkEnd", "RGB", "blockcount", "blocksize", "blockstart")) %>% separate(ID, into= c("gene", "miRNA"), sep=":") %>% filter(gene %in% predMImech)

miRNAtargets_nonConfamnoconsite



```
There are conserved family, non conserved sites between the 2 PAS. 
There are many more results when you include not conserved sites.  

```{r}
miRNAtargets_Otherconsite=  read.table("../data/miRNAbinding/Targets_CS_pctiles.hg19.otherFam.consSite.bed", stringsAsFactors = F,col.names = c("chr", "start", "end", "ID", "score", "strand", "thikStart", "thinkEnd", "RGB", "blockcount", "blocksize", "blockstart")) %>% separate(ID, into= c("gene", "miRNA"), sep=":") %>% filter(gene %in% predMImech)

miRNAtargets_Otherconsite


miRNAtargets_OtherNonconsite=  read.table("../data/miRNAbinding/Targets_CS_pctiles.hg19.otherFam.nonConsSite.bed", stringsAsFactors = F,col.names = c("chr", "start", "end", "ID", "score", "strand", "thikStart", "thinkEnd", "RGB", "blockcount", "blocksize", "blockstart")) %>% separate(ID, into= c("gene", "miRNA"), sep=":") %>% filter(gene %in% predMImech)

miRNAtargets_OtherNonconsite
```

