---
title: "Location of apaQTLs"
author: "Briana Mittleman"
date: "6/14/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this analysis I want to look at the location of the apaQTLs first looking at distance to PAS. Until now I have been using the distance to the peak and have not flipped the strand. This showed me QTLs are close to the PAS but was not the most correct way to do this.  
```{r}
library(workflowr)
library(tidyverse)
library(cowplot)

```

```{r}
PAS=read.table("../data/PAS/APAPAS_GeneLocAnno.5perc.bed",col.names = c("chr", "start", "PASloc", "name", "score", "strand"), stringsAsFactors = F )%>% separate(name, into=c("peakNum", "geneloc"), sep=":") %>% mutate(peak=paste("peak", peakNum, sep="")) %>% select(PASloc, peak)
```
##Distance to PAS  


```{r}
totQTL_bed=read.table("../data/apaQTLs/Total_apaQTLs4pc_5fdr.bed", stringsAsFactors = F, header = T) 
totQTL_txt=read.table("../data/apaQTLs/Total_apaQTLs4pc_5fdr.txt", stringsAsFactors = F, header = T)%>% mutate(name=paste(Gene, Peak, Loc, sep=":")) 
totQTL_strand=totQTL_bed %>% full_join(totQTL_txt, by="name") %>% mutate(strand=ifelse(Strand=="+", "-", "+")) %>% select(SNPchr, SNPstart, SNPend, name, score, strand)

write.table(totQTL_strand, "../data/apaQTLs/Total_apaQTLs4pc_5fdr.WITHSTRAND.bed", quote = F, col.names = T, row.names = F)


nucQTL_bed=read.table("../data/apaQTLs/Nuclear_apaQTLs4pc_5fdr.bed", stringsAsFactors = F, header = T) 
nucQTL_txt=read.table("../data/apaQTLs/Nuclear_apaQTLs4pc_5fdr.txt", stringsAsFactors = F, header = T)%>% mutate(name=paste(Gene, Peak, Loc, sep=":")) 
nucQTL_strand=nucQTL_bed %>% full_join(nucQTL_txt, by="name") %>% mutate(strand=ifelse(Strand=="+", "-", "+")) %>% select(SNPchr, SNPstart, SNPend, name, score, strand)

write.table(nucQTL_strand, "../data/apaQTLs/Nuclear_apaQTLs4pc_5fdr.WITHSTRAND.bed", quote = F, col.names = T, row.names = F)

```

Total:  
```{r}
totQTLs=read.table("../data/apaQTLs/Total_apaQTLs4pc_5fdr.WITHSTRAND.bed",stringsAsFactors = F, header=T)%>%
  separate(name, into=c("gene", "peak", "loc"), sep=":") %>% 
  inner_join(PAS, by="peak") %>% 
  mutate(distance=SNPend-PASloc, dist2PAS=ifelse(strand=="-", -1 *distance, distance))

```


```{r}
ggplot(totQTLs, aes(x=dist2PAS, by=loc, fill=loc)) + geom_histogram(bins=100) 
``` 
Plot by proportion: 

```{r}
ggplot(totQTLs, aes(x=dist2PAS, fill=loc)) + geom_histogram( bins=100) + facet_grid(~loc)
```
Nuclear

```{r}
nucQTLs=read.table("../data/apaQTLs/Nuclear_apaQTLs4pc_5fdr.WITHSTRAND.bed",stringsAsFactors = F, header=T)%>%
  separate(name, into=c("gene", "peak", "loc"), sep=":") %>% 
  inner_join(PAS, by="peak") %>% 
  mutate(distance=SNPend-PASloc, dist2PAS=ifelse(strand=="-", -1 *distance, distance))

```

```{r}
ggplot(nucQTLs, aes(x=dist2PAS, by=loc, fill=loc)) + geom_histogram(bins=100)

```
```{r}
ggplot(nucQTLs, aes(x=dist2PAS, fill=loc)) + geom_histogram( bins=100) + facet_grid(~loc)

```


Plot total and nuclear together: 

```{r}
totalQTLdist=totQTLs %>% select(dist2PAS) %>% mutate(Fraction="Total")
nuclearQTLdist=nucQTLs %>% select(dist2PAS) %>% mutate(Fraction="Nuclear")
bothFractDist=bind_rows(totalQTLdist, nuclearQTLdist)

PASdistplot=ggplot(bothFractDist, aes(x=dist2PAS, fill=Fraction )) + geom_histogram(bins=100,aes(y = (..count..)/sum(..count..))) +labs(y="Proportion of apaQTLs", x="Distance QTL SNP to PAS", title="Distance from QTL SNP to PAS by Fraction") + scale_fill_manual(values=c("deepskyblue3","darkviolet"))  + facet_grid(~Fraction) + theme(text = element_text(size=16),legend.position = "bottom", axis.text.x =element_text(size=14), axis.text.y=element_text(size=14))
```

```{r figure2B, include=FALSE, dev="pdf", fig.height=4, fig.width=10, crop=FALSE}
PASdistplot
```


```{r}
ggplot(bothFractDist, aes(x=dist2PAS, fill=Fraction )) + geom_density() +labs( x="Distance QTL SNP to PAS", title="Distance from QTL SNP to PAS") + scale_fill_manual(values=c("deepskyblue3","darkviolet")) + facet_grid(~Fraction) + theme(legend.position = "none")
  
```

##Metagene plot  

I want to plot by normalized position in the gene.  

```{r}
genes=read.table("../../genome_anotation_data/refseq.ProteinCoding.bed",col.names = c("chrom", "Genestart", "Geneend", "gene", "score", "strand") ,stringsAsFactors = F) %>% select(Genestart, Geneend, gene)
```

Total: 

```{r}
totQTLs_gene= totQTLs %>% inner_join(genes, by="gene")%>% mutate(geneLength=Geneend-Genestart) %>% mutate(dist2QTL=ifelse(strand=="+",as.numeric(SNPend)-as.numeric(Genestart), as.numeric(Geneend)-as.numeric(SNPend)), propGene=dist2QTL/geneLength)  %>% filter(propGene>-3 & propGene<3)

```

```{r}
ggplot(totQTLs_gene, aes(x=propGene, fill=loc)) + geom_histogram(bins=50)  + labs(x="Proportion of gene body", y="number QTLs", title="Total apaQTLs") + geom_vline(xintercept =0,color= "black") + geom_vline(xintercept =1,color= "black")
```
```{r}
totQTLs_gene_intron_utr= totQTLs_gene %>% filter(loc=="intron"| loc =="utr3")
ggplot(totQTLs_gene_intron_utr, aes(x=propGene, fill=loc)) + geom_density()  + labs(x="Proportion of gene body", title="Total apaQTLs") + geom_vline(xintercept =0,color= "black") + geom_vline(xintercept =1,color= "black")

```


There are about 48 QTLs outside. 

I can look at only those in the gene body:

```{r}
totQTLs_gene_body= totQTLs_gene %>% filter(propGene>=0, propGene<=1)
  
ggplot(totQTLs_gene_body, aes(x=propGene, fill=loc)) + geom_histogram(bins=50)  + labs(x="Proportion of gene body", y="number QTLs", title="Total apaQTLs in gene body") 

```

There are 181 in the gene body

Nuclear:
```{r}
nucQTLs_gene= nucQTLs %>% inner_join(genes, by="gene")%>% mutate(geneLength=Geneend-Genestart)%>% mutate(dist2QTL=ifelse(strand=="+",as.numeric(SNPend)-as.numeric(Genestart), as.numeric(Geneend)-as.numeric(SNPend)), propGene=dist2QTL/geneLength)  %>% filter(propGene>-3 & propGene<3)

```


```{r}
ggplot(nucQTLs_gene, aes(x=propGene, fill=loc)) + geom_histogram(bins=50)  + labs(x="Proportion of gene body", y="number QTLs", title="Nuclear apaQTLs") + geom_vline(xintercept =0,color= "black") + geom_vline(xintercept =1,color= "black")

```

```{r}

nucQTLs_gene_intron_utr= nucQTLs_gene %>% filter(loc=="intron"| loc =="utr3")
ggplot(nucQTLs_gene_intron_utr, aes(x=propGene, fill=loc)) + geom_density()  + labs(x="Proportion of gene body", title="Nuclear apaQTLs") + geom_vline(xintercept =0,color= "black") + geom_vline(xintercept =1,color= "black")

```

there are 77 outside of 500% of gene body 


```{r}
nucQTLs_gene_body= nucQTLs_gene %>% filter(propGene>=0, propGene<=1)

ggplot(nucQTLs_gene_body, aes(x=propGene, fill=loc)) + geom_histogram(bins=50)  + labs(x="Proportion of gene body", y="number QTLs", title="Nuclear apaQTLs in gene body") 

```
334 are in the gene body.  

Plot both togther:


```{r}
nucQTLs_geneprop= nucQTLs_gene %>% select(propGene) %>% mutate(Fraction="Nuclear")
totQTLs_geneprop= totQTLs_gene %>% select(propGene) %>% mutate(Fraction="Total")
genepropboth=bind_rows(totQTLs_geneprop,nucQTLs_geneprop)

ggplot(genepropboth, aes(x=propGene,fill=Fraction)) + geom_histogram(bins=100) + geom_vline(xintercept =0,color= "black") + geom_vline(xintercept =1,color= "black")+ scale_fill_manual(values=c("deepskyblue3","darkviolet"))  + labs(x="Proportion of gene body", y="Number of apaQTLs", title="Metagene plot for apaQTL SNP location")
```
Density plot

```{r}
propgenebodyplot=ggplot(genepropboth, aes(x=propGene,fill=Fraction)) + geom_histogram(bins=100,aes(y = (..count..)/sum(..count..))) + geom_vline(xintercept =0,color= "black") + geom_vline(xintercept =1,color= "black")+ scale_fill_manual(values=c("deepskyblue3","darkviolet"))  + labs(y="Proportion of qtls",x="Proportion of gene body", title="Metagene plot for apaQTL SNP location")+  theme(text = element_text(size=16),legend.position = "bottom", axis.text.x =element_text(size=14), axis.text.y=element_text(size=14)) + facet_grid(~Fraction)
propgenebodyplot
```
```{r figure2C, include=FALSE, dev="pdf", fig.height=4, fig.width=10, crop=FALSE}
propgenebodyplot
```


```{r}
geneprop_nuc=genepropboth %>% filter(Fraction=="Nuclear")

ggplot(geneprop_nuc, aes(x=propGene)) + geom_density(bins=100, fill="#7570B3") + geom_vline(xintercept =0,color= "black") + geom_vline(xintercept =1,color= "black")  + labs(x="Proportion of gene body", title="apaQTL SNP location")+  theme(text = element_text(size=16), axis.text.x =element_text(size=14), axis.text.y=element_text(size=14))
```


