---
title: "Read distribution Deep tools plots"
author: "Briana Mittleman"
date: "4/22/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this analysis I will create the read distribution figures. These are created using deeptools. I have merged total and nuclear bam files from the read mapping pipeline. I will convert these to bigwigs in order to map the reads against features with deeptools. 


##Create BW files  
```{bash,eval=F}
mkdir ../data/mergedBW_byfrac
mkdir ../data/DTmatrix
mkdir ../output/dtPlots

module load Anaconda3 
source activate three-prime-env

sbatch bam2bw.sh ../data/mergedbyFracBam/Total.SamplesMerged.sort.bam ../data/mergedBW_byfrac/Total.SamplesMerged.bw 

sbatch bam2bw.sh ../data/mergedbyFracBam/Nuclear.SamplesMerged.sort.bam ../data/mergedBW_byfrac/Nuclear.SamplesMerged.bw
```

##Map along gene bodies  


```{bash,eval=F}
sbatch BothFracDTPlotGeneRegions.sh
```


##Redo with Normalized RPKM  

I need to create normalized bw from each bam file. I then merge by fraction and convert back to bigwig from bedgraph. 

I added a rule to the first snakefile that created the normalized files. Then I run the following to merge and create the bw.  

```{bash,eval=F}
sbatch mergeBW_norm.sh
```
Next I create the plot with deeptools.  

```{bash,eval=F}
sbatch BothFracDTPlotGeneRegions_normalized.sh
```



##Same plots with RNA seq  

 I combined 6 samples of the Geuvadis RNA seq into one file for this. 
```{bash,eval=F}
sbatch RNAseqDTplot.sh
```


##Zoom in on the TES  

I will use the non normalized version of the merged bigwig and plot the 100bp up and downstream of annoated TES using bedtools. I will also do this in our RNA data.  

BW total/nuclear: 
../data/mergedBW_byfrac/Total.SamplesMerged.bw 
../data/mergedBW_byfrac/Nuclear.SamplesMerged.bw

RNA: 
../data/GeuvadisRNA/RNAseqGeuvadis_STAR_6samp_MergedBams.sort.bw 

TES:  
/project2/gilad/briana/genome_anotation_data/ncbiRefSeq_endProtCodGenes_sort.bed

```{bash,eval=F}
sbatch TESplots100bp.sh
sbatch TESplots150bp.sh
sbatch TESplots200bp.sh
```

I want to only look at the first base in each read. This is the actual PAS. I can recompute the BW with the  â€“Offset argument.  

```{bash,eval=F}
sbatch bam2BW_5primemost.sh ../data/mergedbyFracBam/Total.SamplesMerged.sort.bam ../data/mergedBW_byfrac/Total.SamplesMerged.5primemost.bw 

sbatch bam2BW_5primemost.sh ../data/mergedbyFracBam/Nuclear.SamplesMerged.sort.bam ../data/mergedBW_byfrac/Nuclear.SamplesMerged.5primemost.bw 
```


