---
title: "Relationship between RNA, net-seq, h3k27ac"
author: "Briana Mittleman"
date: "5/15/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this analysis I want to look at the correlation between the net seq daata, rna seq, 4su, and h3k27ac to understand the relationship between nascent transcription and steady state RNA. This will be similar to the analysis in Li et al 2016 figure 1c.  

```{r}
library(tidyverse)
library(gplots)
library(gdata)
library(reshape2)
library(workflowr)
```

##H3K27ac at TSS  


```{r}
h3k27ac=read.table("../data/h3k27ac/H3K27acShyam_TSS_fixed.txt", header = T,stringsAsFactors = F)%>% dplyr::select(-Chr, -Start, -End,-Strand, -Length) %>% dplyr::rename("ID"= Geneid)
```


##RNA seq


```{r}
RNA=read.table("../data/fourSU/tr_decay_table_norm.txt", header=T, stringsAsFactors = F)%>%  dplyr::select(gene_id,contains("RNAseq_14000")) %>%  dplyr::rename("ID"=gene_id)
```
I also have the kalisto TPM that I can try:  

```{r}
RNA_TPM=read.table('../data/RNAseq/kallisto_RNAseq.txt', stringsAsFactors = F,header = T) %>% dplyr::rename("ID"=gene)
```


##4su  

```{r}
fourSU=read.table("../data/fourSU/tr_decay_table_norm.txt", header=T, stringsAsFactors = F)%>%  dplyr::select(gene_id,contains("4su_30")) %>% dplyr::rename("ID"=gene_id)
```

tpm 4su

```{r}
foursu_tpm=read.table("../data/fourSU/kallisto_4sU.txt", header = T, stringsAsFactors = F) %>% dplyr::rename("ID"=gene)
```


##Netseq  

I want to quantify reads 1kb on either side of the TSS. I will use the gencode v19 annotations to match the files above. I need to convert the gtf file into an saf file with the TSS.  

```{bash,eval=F}
python makegencondeTSSfile.py
```
Run feature counts with the 16 net seq libraries and this TSS file.  

```{bash,eval=F}
sbatch netseqFC.sh
```
Fix header 
```{bash,eval=F}
python fixFChead_bothfrac.py ../data/netseq/netseq_TSS.fc ../data/netseq/netseq_TSS.fixed.fc
```


```{r}
netseq=read.table("../data/netseq/netseq_TSS.fixed.fc", stringsAsFactors = F, header = T) %>% dplyr::select(-Chr, -Strand, -Start, -Length, -End) %>% dplyr::rename("ID"=Geneid)
```

###Total:

Genes to include: 
```{r}
sig_genes=read.table(file="../data/highdiffsiggenes.txt",col.names = "GeneName")
```

I will have to change the gene names for the 3' info:  
```{r}
geneNames=read.table("../../genome_anotation_data/ensemble_to_genename.txt", sep="\t", col.names = c('geneid', 'GeneName', 'source' ),stringsAsFactors = F) %>% dplyr::select(-source)
```

```{r}
peaknumlist=read.table("../data/peaks_5perc/APApeak_Peaks_GeneLocAnno.5perc.bed", stringsAsFactors = F, header=F, col.names = c("chr", "start","end", "id", "score", "strand"))  %>% separate(id, into=c("peaknum", "geneid"), sep=":") %>% mutate(peakid=paste("peak", peaknum,sep=""))

TotAPA=read.table("../data/peakCoverage/APAPeaks.ALLChrom.Filtered.Named.GeneLocAnnoPARSED.Total.Quant.Fixed.fc", stringsAsFactors = F, header = T) %>%  dplyr::select(-Chr, -Start, -End, -Strand, -Length) %>% separate(Geneid, into=c("peakid","chrom", "start", "end", "strand", "geneID"),sep=":")  %>% separate(geneID,into=c("GeneName", "loc"),sep="_") %>% inner_join(sig_genes, by="GeneName")%>%  dplyr::semi_join(peaknumlist, by="peakid") %>%  dplyr::select(-chrom , -start, -end, -strand, -loc)

TotApaMelt=melt(TotAPA, id.vars =c( "peakid", "GeneName"), value.name="count", variable.name="Ind") %>% separate(Ind, into=c('Individual', 'fraction') ,sep="_")%>% dplyr::select(peakid, GeneName, Individual, count) %>% inner_join(geneNames,by="GeneName") %>% group_by(Individual,geneid) %>% summarize(TotApa=sum(count)) %>% ungroup() %>% dplyr::rename("ID"=geneid)  %>% mutate(Individual=paste("TotAPA_", Individual, sep=""))

##spread

totApaSpread= spread(TotApaMelt, Individual,TotApa)
```

###Nuclear  
```{r}

NucAPA=read.table("../data/peakCoverage/APAPeaks.ALLChrom.Filtered.Named.GeneLocAnnoPARSED.Nuclear.Quant.Fixed.fc", stringsAsFactors = F, header = T) %>%  dplyr::select(-Chr, -Start, -End, -Strand, -Length) %>% separate(Geneid, into=c("peakid","chrom", "start", "end", "strand", "geneID"),sep=":")  %>% separate(geneID,into=c("GeneName", "loc"),sep="_") %>% inner_join(sig_genes, by="GeneName")%>%  dplyr::semi_join(peaknumlist, by="peakid") %>%  dplyr::select(-chrom , -start, -end, -strand, -loc)


NucApaMelt=melt(NucAPA, id.vars =c( "peakid", "GeneName"), value.name="count", variable.name="Ind") %>% separate(Ind, into=c('Individual', 'fraction') ,sep="_")%>% dplyr::select(peakid, GeneName, Individual, count) %>% inner_join(geneNames,by="GeneName") %>% group_by(Individual,geneid) %>% summarize(NucApa=sum(count)) %>% ungroup() %>% dplyr::rename("ID"=geneid)  %>% mutate(Individual=paste("NucAPA_", Individual, sep=""))

##spread

nucApaSpread= spread(NucApaMelt, Individual,NucApa)
```

##Correlation:

I will join all of these based on the genes we have data for in all. 

```{r}
#4su, h3k27, RNA, netseq, nuc, total
allpheno=fourSU %>% dplyr::inner_join(h3k27ac, by="ID") %>% dplyr::inner_join(RNA, by="ID") %>% dplyr::inner_join(netseq, by="ID") %>%  dplyr::inner_join(nucApaSpread, by="ID") %>%  dplyr::inner_join(totApaSpread, by="ID") 
```

```{r}
allpheno_matrix= as.matrix(allpheno %>% dplyr::select(-ID))
```

```{r}

my_palette <- colorRampPalette(c("white", "yellow", "orange", "red", "black"))(n = 100)
allphenocorr= abs(round(cor(allpheno_matrix,method="spearman"),2))
##4su-red, h3k27-green, RNA-blue, netseq-purple, nuc-orange, total-yellow
colBar=c(rep("Red",65), rep("Green", 59), rep("Blue",69), rep("Purple", 16),rep("Orange", 52),rep("Yellow", 52))
heatmap.2(as.matrix(allphenocorr),trace="none", dendrogram='col',ColSideColors=colBar, col=my_palette)
          
```

##Pairwise graphs averaging accross indviduals:
First I will take the mean for all individuals for each phenotype:  


```{r}
h3k27ac_mean=melt(h3k27ac,id.vars = "ID") %>% group_by(variable) %>% mutate(sumInd=sum(value)) %>% ungroup() %>% mutate(normVal=value/sumInd) %>% group_by(ID) %>% summarize(H3K27AC=mean(normVal))%>% filter(H3K27AC!=0)

RNA_mean=melt(RNA,id.vars = "ID") %>% group_by(ID) %>% summarize(Rna=mean(value)) %>% filter(Rna!=0)

foursu_mean= melt(fourSU,id.vars = "ID") %>%  group_by(variable) %>% mutate(sumInd=sum(value)) %>% ungroup() %>% mutate(normVal=value/sumInd) %>% group_by(ID) %>% summarize(FourSU=mean(normVal))%>% filter(FourSU!=0)
  

netseq_mean= melt(netseq,id.vars = "ID") %>% group_by(variable) %>% mutate(sumInd=sum(value)) %>% ungroup() %>% mutate(normVal=value/sumInd) %>% group_by(ID) %>% summarize(NetSeq=mean(normVal))%>% filter(NetSeq!=0)

totapa_mean= melt(totApaSpread,id.vars = "ID") %>% group_by(variable) %>% mutate(sumInd=sum(value)) %>% ungroup() %>% mutate(normVal=value/sumInd) %>% group_by(ID) %>% summarize(TotApa=mean(normVal))%>% filter(TotApa!=0)


nucapa_mean= melt(nucApaSpread,id.vars = "ID") %>% group_by(variable) %>% mutate(sumInd=sum(value)) %>% ungroup() %>% mutate(normVal=value/sumInd) %>% group_by(ID) %>% summarize(NucApa=mean(normVal))%>% filter(NucApa!=0)

```

Join all of these:

```{r}
Allpheno_mean= h3k27ac_mean %>% inner_join(RNA_mean,by="ID") %>% inner_join(foursu_mean, by="ID") %>%   inner_join(netseq_mean, by="ID") %>% inner_join(totapa_mean, by="ID")  %>%inner_join(nucapa_mean, by="ID")

```


```{r}
ggplotRegression <- function (fit) {

require(ggplot2)

ggplot(fit$model, aes_string(x = names(fit$model)[2], y = names(fit$model)[1])) + 
  geom_point() +
  stat_smooth(method = "lm", col = "red") +
  labs(title = paste("Adj R2 = ",signif(summary(fit)$adj.r.squared, 5),
                     "Intercept =",signif(fit$coef[[1]],5 ),
                     " Slope =",signif(fit$coef[[2]], 5),
                     " P =",signif(summary(fit)$coef[2,4], 5)))
}
```


Plots:

```{r}

ggplotRegression(lm(log10(Allpheno_mean$NetSeq)~ log10(Allpheno_mean$Rna)))
```

```{r}
ggplotRegression(lm(log10(Allpheno_mean$FourSU)~ log10(Allpheno_mean$Rna)))
```

```{r}
ggplotRegression(lm(log10(Allpheno_mean$H3K27AC)~ log10(Allpheno_mean$Rna)))
```
```{r}
ggplotRegression(lm(log10(Allpheno_mean$H3K27AC)~ log10(Allpheno_mean$NetSeq)))
```
```{r}
ggplotRegression(lm(log10(Allpheno_mean$H3K27AC)~ log10(Allpheno_mean$FourSU)))
```

```{r}
ggplotRegression(lm(log10(Allpheno_mean$FourSU)~ log10(Allpheno_mean$NetSeq)))
```

```{r}
ggplotRegression(lm(log10(Allpheno_mean$Rna)~ log10(Allpheno_mean$NetSeq)))
```

```{r}
ggplotRegression(lm(log10(Allpheno_mean$H3K27AC)~ log10(Allpheno_mean$NetSeq)))
```

```{r}
ggplotRegression(lm(log10(Allpheno_mean$H3K27AC)~ log10(Allpheno_mean$TotApa)))
```

```{r}
ggplotRegression(lm(log10(Allpheno_mean$H3K27AC)~ log10(Allpheno_mean$NucApa)))
```


```{r}
ggplotRegression(lm(Allpheno_mean$NucApa~ Allpheno_mean$Rna))
ggplotRegression(lm(Allpheno_mean$NucApa~ Allpheno_mean$FourSU))
```
```{r}
ggplotRegression(lm(Allpheno_mean$TotApa~ Allpheno_mean$Rna))
ggplotRegression(lm(Allpheno_mean$TotApa~ Allpheno_mean$FourSU))
```
heatmap correlation for these:

```{r}
Allpheno_mean_mat= as.matrix(Allpheno_mean %>% dplyr::select(-ID))
Allpheno_mean_matcorr= abs(round(cor(Allpheno_mean_mat,method="spearman"),2))


Allpheno_mean_matcorrOrder=as.data.frame(Allpheno_mean_matcorr) %>% select(NetSeq, H3K27AC,NucApa,TotApa,FourSU,Rna)


heatmap.2(as.matrix(Allpheno_mean_matcorrOrder),trace="none",dendrogram = "none", col=my_palette,Colv = F)
```

```{r}
Allpheno_mean_matcorrOrder
```


remove net and k27ac 

```{r}
Allpheno_mean_small=as.matrix(Allpheno_mean %>% dplyr::select(-ID,-H3K27AC,-NetSeq))
Allpheno_meanSM_matcorr= abs(round(cor(Allpheno_mean_small,method="spearman"),2))

heatmap.2(as.matrix(Allpheno_meanSM_matcorr),trace="none", col=my_palette, ,dendrogram = "none")
```

