---
title: "Signal Site analysis"
author: "Briana Mittleman"
date: "4/23/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(seqLogo)
library(tidyverse)
library(RColorBrewer)
library(workflowr)
library(reshape2)
library(gplots)
library(cowplot)
```

In this analysis I will plot the distribution of signal sites upstream of the PAS I have found. 


First I use a python script to make a bed file with the 100 base pairs upsream of the PAS:

```{bash,eval=F}
module load Anaconda3
source activate three-prime-env
mkdir ../data/SignalSiteFiles
python Upstream100Bases_general.py ../data/PAS/APAPAS_GeneLocAnno.5perc.bed ../data/SignalSiteFiles/APAPAS_100up.bed
```


Now I use bedtools nuc to get the sequence for each of these regions:  
```{bash,eval=F}
sbatch getSeq100up.sh 
```

I can now run the DistPAS2Sig.py which will give me the location for the signal site for each PAS.I am running this with the 12 most common PAS signal sites. 

```{bash,eval=F}
sbatch run_distPAS2Sig.sh
```

Upload all of the results:

```{r}
Loc_AATAAA= read.table("../data/SignalSiteFiles/Loc_AATAAA_Distance2end.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AATAAA")
Loc_AAAAAG= read.table("../data/SignalSiteFiles/Loc_AAAAAG_Distance2end.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AAAAAG")
Loc_AATACA= read.table("../data/SignalSiteFiles/Loc_AATACA_Distance2end.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AATACA")
Loc_AATAGA= read.table("../data/SignalSiteFiles/Loc_AATAGA_Distance2end.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AATAGA")
Loc_AATATA= read.table("../data/SignalSiteFiles/Loc_AATATA_Distance2end.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AATATA")
Loc_ACTAAA= read.table("../data/SignalSiteFiles/Loc_ACTAAA_Distance2end.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="ACTAAA")
Loc_AGTAAA= read.table("../data/SignalSiteFiles/Loc_AGTAAA_Distance2end.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AGTAAA")
Loc_ATTAAA= read.table("../data/SignalSiteFiles/Loc_ATTAAA_Distance2end.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="ATTAAA")
Loc_CATAAA= read.table("../data/SignalSiteFiles/Loc_CATAAA_Distance2end.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="CATAAA")
Loc_GATAAA= read.table("../data/SignalSiteFiles/Loc_GATAAA_Distance2end.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="GATAAA")
Loc_TATAAA= read.table("../data/SignalSiteFiles/Loc_TATAAA_Distance2end.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="TATAAA")
Loc_AAAAAA= read.table("../data/SignalSiteFiles/Loc_AAAAAA_Distance2end.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AAAAAA")
```

Join these together:  

```{r}
AllsiteDF=as.data.frame(rbind(Loc_AATAAA,Loc_AAAAAG,Loc_AATACA,Loc_AATAGA,Loc_AATATA,Loc_ACTAAA,Loc_AGTAAA,Loc_ATTAAA, Loc_GATAAA,Loc_TATAAA,Loc_CATAAA, Loc_AAAAAA))
```
```{r}
colourCount = length(unique(AllsiteDF$Site))
getPalette = colorRampPalette(brewer.pal(8, "Set1"))
```



```{r}

AllsiteDF_sep = AllsiteDF %>% separate(PAS, int=c("GenePeak", "Location"), sep="_")
ggplot(AllsiteDF_sep, aes(x=Distance2PAS, by=Site, col=Site)) + stat_ecdf() + facet_wrap(~Location)


```

Check to see if any PAS have more than one signal site detected:

```{r}
AllsiteDFmultsites=AllsiteDF %>% group_by(PAS) %>% mutate(nSites=n()) %>% filter(nSites>1)
```
First take the perfect match within 50 bp then use the closest. 

Write out the AllSite in order to use it in the chooseSignalSite.py script:  

```{r}
write.table(AllsiteDF, file="../data/SignalSiteFiles/AllSignalSite.txt", quote=F, col.names = F, row.names = F, sep="\t")
```


```{bash, eval=F}
python chooseSignalSite.py ../data/SignalSiteFiles/AllSignalSite.txt ../data/SignalSiteFiles/AllSignalSite_1perPAS.txt
```

```{r}
AllsiteDF_1per=read.table(file="../data/SignalSiteFiles/AllSignalSite_1perPAS.txt", col.names = colnames(AllsiteDF)) %>% mutate(NegCount=-1*as.integer(as.character(Distance2PAS)))
```

Plot

```{r}
dist2signalsiteplot=ggplot(AllsiteDF_1per, aes(group=Site, x=NegCount, fill=Site)) + geom_histogram(position="stack",bins=50 ) + labs(x="Distance from PAS", y="N annotated Sites", title="Location of annotated signal sites")  +  scale_fill_manual(values = getPalette(colourCount))
dist2signalsiteplot

ggsave(dist2signalsiteplot, file="../output/SignalSitePlot.png")
```
Plot with proportion:  
```{r}
allPAS=read.table("../data/PAS/APAPAS_GeneLocAnno.5perc.bed", stringsAsFactors = F, col.names = c("chr","start","end","PAS","score","strand"))

AllsiteDF_1per_prop= AllsiteDF_1per %>% group_by(Site,NegCount) %>% summarise(CountperPos=n()) %>% mutate(TotCount=sum(CountperPos),prop=CountperPos/nrow(allPAS))
```

Plot with prop:
```{r}
dist2signalsiteplotprop=ggplot(AllsiteDF_1per_prop, aes(group=Site, x=NegCount,y=prop, fill=Site)) + geom_histogram(position="stack",bins=50,stat="identity" ) + labs(x="Distance from PAS", y="Proportion of annotated Sites", title="Location of annotated signal sites",fill="Site Sequence") +  scale_fill_manual(values = getPalette(colourCount))+ theme(text = element_text(size=20, face="bold"),axis.text.x = element_text(size = 16),axis.text.y = element_text(size = 16), legend.position = "bottom",plot.title = element_text(size=22))+ guides(fill=guide_legend(nrow=3,byrow=TRUE))
dist2signalsiteplotprop
```
```{r}
nrow(AllsiteDF_1per)/nrow(allPAS) 
```


Seperate by location:

```{r}
AllsiteDF_1per_sep= AllsiteDF_1per %>%separate(PAS, int=c("GenePeak", "Location"), sep="_")
```

```{r}
dist2signalsiteplot_byloc=ggplot(AllsiteDF_1per_sep, aes(group=Site, x=NegCount, fill=Site)) + geom_histogram(position="stack",bins=50 ) + labs(x="Distance from PAS", y="N annotated Sites", title="Location of annotated signal sites") + facet_wrap(~Location)+  scale_fill_manual(values = getPalette(colourCount))

dist2signalsiteplot_byloc

ggsave(dist2signalsiteplot_byloc, file="../output/SignalSitePlotbyLoc.png")
```
Proportion:  
```{r}
allPAS_byloc=allPAS %>% separate(PAS,into=c("peakid", "loc"),sep="_") %>% group_by(loc) %>% summarise(nLoc=n())

allPAS_byloc_new=as.data.frame(allPAS_byloc$nLoc %>% t())
colnames(allPAS_byloc_new) = allPAS_byloc$loc



AllsiteDF_1per_sep_INTRON=AllsiteDF_1per_sep %>% filter(Location=="intron") %>%  group_by(Site,NegCount) %>% summarise(CountperPos=n()) %>% mutate(TotCount=sum(CountperPos),prop=CountperPos/(allPAS_byloc_new$intron)) %>% mutate(Cononical=ifelse(Site=="AATAAA", "AATAAA", ifelse(Site=="ATTAAA", "AATTAA", "Other")))

ggplot(AllsiteDF_1per_sep_INTRON, aes(group=Cononical, x=NegCount,y=prop, fill=Cononical)) + geom_histogram(position="stack",bins=5, stat="identity" ) + labs(x="Distance from PAS", y="Propotion of annotated Sites", title="Location of annotated signal sites \nfor Intronic PAS", caption = "Other: AAAAAA, AAAAAG, AATACA, AATAGA,AATATA, ACTAAA, AGTAAA,CATAAA, GATAAA,TATAAA") +  scale_fill_manual(values = getPalette(colourCount))+  theme(text = element_text(size=20, face="bold"),axis.text.x = element_text(size = 16), legend.position = "bottom", axis.text.y = element_text(size = 16),plot.title = element_text(size=22),plot.caption = element_text(hjust = 0,size=10))
```
```{r}
AllsiteDF_1per_sep_UTR=AllsiteDF_1per_sep %>% filter(Location=="utr3") %>%  group_by(Site,NegCount) %>% summarise(CountperPos=n()) %>% mutate(TotCount=sum(CountperPos),prop=CountperPos/(allPAS_byloc_new$intron)) %>% mutate(Cononical=ifelse(Site=="AATAAA", "AATAAA", ifelse(Site=="ATTAAA", "AATTAA", "Other")))

ggplot(AllsiteDF_1per_sep_UTR, aes(group=Cononical, x=NegCount,y=prop, fill=Cononical)) + geom_histogram(position="stack", stat="identity" ) + labs(x="Distance from PAS", y="Propotion of annotated Sites", title="Location of annotated signal sites \nin UTR",caption = "Other: AAAAAA, AAAAAG, AATACA, AATAGA,AATATA, ACTAAA, AGTAAA,CATAAA, GATAAA,TATAAA") +  scale_fill_manual(values = getPalette(colourCount)) + theme(text = element_text(size=20, face="bold"),axis.text.x = element_text(size = 16), legend.position = "bottom", axis.text.y = element_text(size = 16),plot.title = element_text(size=22),plot.caption = element_text(hjust = 0,size=10))+ guides(fill=guide_legend(nrow=1,byrow=TRUE))
```

```{r}
Propwith=c(nrow(AllsiteDF_1per_sep %>% filter(Location=="intron"))/allPAS_byloc_new$intron,nrow(AllsiteDF_1per_sep %>% filter(Location=="utr3"))/allPAS_byloc_new$utr3,nrow(AllsiteDF_1per_sep %>% filter(Location=="utr5"))/allPAS_byloc_new$utr5,nrow(AllsiteDF_1per_sep %>% filter(Location=="cds"))/allPAS_byloc_new$cds,nrow(AllsiteDF_1per_sep %>% filter(Location=="end"))/allPAS_byloc_new$end)
Locations=c("intron", "utr3", "utr5", "Coding", "Downstream")
propdf=as.data.frame(cbind(Location=Locations,Proportion=Propwith))
propdf$Proportion=as.numeric(as.character(propdf$Proportion))

all=ggplot(propdf,aes(x=Location, y=Proportion, fill=Location)) + geom_bar(stat="identity") + theme(axis.text.x = element_text(angle = 90, hjust = 1))
all
```

```{r}
AllsiteDF_1per_sep_noncon=AllsiteDF_1per_sep %>% filter(Site != "AATAAA")

Propwithnotcon=c(nrow(AllsiteDF_1per_sep_noncon %>% filter(Location=="intron"))/allPAS_byloc_new$intron,nrow(AllsiteDF_1per_sep_noncon %>% filter(Location=="utr3"))/allPAS_byloc_new$utr3,nrow(AllsiteDF_1per_sep_noncon %>% filter(Location=="utr5"))/allPAS_byloc_new$utr5,nrow(AllsiteDF_1per_sep_noncon %>% filter(Location=="cds"))/allPAS_byloc_new$cds,nrow(AllsiteDF_1per_sep_noncon %>% filter(Location=="end"))/allPAS_byloc_new$end)
Locations=c("intron", "utr3", "utr5", "Coding", "Downstream")
propdf_noncon=as.data.frame(cbind(Location=Locations,Proportion=Propwithnotcon))
propdf_noncon$Proportion=as.numeric(as.character(propdf_noncon$Proportion))

non=ggplot(propdf_noncon,aes(x=Location, y=Proportion, fill=Location)) + geom_bar(stat="identity")+theme(axis.text.x = element_text(angle = 90, hjust = 1))
non
```
```{r}
AllsiteDF_1per_sep_con=AllsiteDF_1per_sep %>% filter(Site == "AATAAA")

Propwithcon=c(nrow(AllsiteDF_1per_sep_con %>% filter(Location=="intron"))/allPAS_byloc_new$intron,nrow(AllsiteDF_1per_sep_con %>% filter(Location=="utr3"))/allPAS_byloc_new$utr3,nrow(AllsiteDF_1per_sep_con %>% filter(Location=="utr5"))/allPAS_byloc_new$utr5,nrow(AllsiteDF_1per_sep_con %>% filter(Location=="cds"))/allPAS_byloc_new$cds,nrow(AllsiteDF_1per_sep_con %>% filter(Location=="end"))/allPAS_byloc_new$end)
Locations=c("intron", "utr3", "utr5", "Coding", "Downstream")
propdf_con=as.data.frame(cbind(Location=Locations,Proportion=Propwithcon))
propdf_con$Proportion=as.numeric(as.character(propdf_con$Proportion))

con=ggplot(propdf_con,aes(x=Location, y=Proportion, fill=Location)) + geom_bar(stat="identity")+ theme(axis.text.x = element_text(angle = 90, hjust = 1))
con
```


```{r}
plot_grid(all, con, non, labels=c("All PAS", "Cononical PAS", "Non-conical PAS"))
```


##Strong evidence PAS  

For future analysis I want to have a set of PAS with evidence for a signal site.  I want those signal sites upstream 10-50 basepairs.

```{r}
AllsiteDF_1per_use= AllsiteDF_1per %>% filter(Distance2PAS>10, Distance2PAS<50) %>% separate(PAS,into=c("peakid", "loc"),sep="_") %>% separate(peakid,into=c("Peaknum", "gene"),sep=":") %>% mutate(PAS=paste("peak", Peaknum, sep="")) %>% dplyr::rename("UpstreamDist"=NegCount) %>%  select(PAS, gene, loc, Site, UpstreamDist)
``` 


```{r}
ggplot(AllsiteDF_1per_use, aes(x=loc)) + geom_histogram(stat="count")
```

Write these out for 
```{r}
write.table(AllsiteDF_1per_use, file="../data/PAS/PASwSignalSite.txt", col.names = T, row.names = F, quote = F, sep="\t")
```



##Plot color cononical vs non cononical:  

```{r}

AllsiteDF_1per_prop_col= AllsiteDF_1per_prop %>% mutate(Cononical=ifelse(Site=="AATAAA", "AATAAA", ifelse(Site=="ATTAAA", "AATTAA", "Other")))

dist2signalsiteplotprop=ggplot(AllsiteDF_1per_prop_col, aes(group=Cononical, x=NegCount,y=prop, fill=Cononical)) + geom_histogram(position="stack",bins=50,stat="identity" ) + labs(x="Distance from PAS", y="Proportion of annotated Sites", title="Location of annotated signal sites",fill=" ", caption = "Other: AAAAAA, AAAAAG, AATACA, AATAGA,AATATA, ACTAAA, AGTAAA,CATAAA, GATAAA,TATAAA") +   theme(text = element_text(size=20, face="bold"),axis.text.x = element_text(size = 16), legend.position = "bottom", axis.text.y = element_text(size = 16),plot.title = element_text(size=22),plot.caption = element_text(hjust = 0,size=10))+ guides(fill=guide_legend(nrow=1,byrow=TRUE))  +  scale_fill_manual(values = getPalette(colourCount))
dist2signalsiteplotprop 


#scale_fill_discrete(name="Site", labels=c("AATAAA","ATTAAA", "AAAAAA, AAAAAG, AATACA, AATAGA, AATATA, ACTAAA, AGTAAA,CATAAA, GATAAA,TATAAA")
```


```{r figure1bMain, include=FALSE, dev="pdf", fig.height=9, fig.width=9, crop=FALSE}
dist2signalsiteplotprop
```



Enrichment in intron

N intronic that have a PAS:
```{r}
propwSS_intron=AllsiteDF_1per_use %>% filter(loc=="intron") %>% nrow()

#nuclear PAS
intronPAS=read.table("../data/PAS/APAPAS_GeneLocAnno.5perc.bed", col.names = c("chr","start","end", "id", "score", "strand")) %>% separate(id, into=c("pas", "loc"), sep="_") %>% filter(loc=="intron") %>% nrow()


withSS_intron=propwSS_intron/intronPAS
withSS_intron
```


```{r}
intronannotation=read.table("/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_intron.sort.bed", col.names = c("chr", "start", "end", "loc", "gene", "score", "strand"))%>% mutate(name=paste(gene, loc, strand, sep="_")) %>% select(chr, start, end, name, score, strand)



write.table(intronannotation, "/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_intronNamed.sort.bed", col.names = F, row.names = F,quote = F, sep="\t")
```


Compare this to a background: 

I need 40 basepair regions in introns. 

```{bash,eval=F}
bedtools makewindows -i src -b /project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_intronNamed.sort.bed -w 40 > /project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_intron.sort_randomIntervals.bed
```
Make this into a bed file (with strand):  

```{bash,eval=F}
python fix_randomIntron.py
```



I need to get the sequences for these with bedtools nuc.  

```{bash,eval=F}
bedtools nuc -s -seq -fi /project2/gilad/briana/genome_anotation_data/genome/Homo_sapiens.GRCh37.75.dna_sm.all.fa -bed /project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_intron.sort_randomIntervals.fixed.bed > /project2/gilad/briana/apaQTL/data/SignalSiteFiles/ncbiRefSeq_intron.sort_randomIntervalsSeq.bed
```



```{bash,eval=F}
sbatch run_dist2sig_randomintron.sh
```


Upload all of the results:

```{r}
Loc_AATAAA_randomIntron= read.table("../data/SignalSiteFiles/Loc_AATAAA_Distance2end_randomIntron.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AATAAA")
Loc_AAAAAG_randomIntron= read.table("../data/SignalSiteFiles/Loc_AAAAAG_Distance2end_randomIntron.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AAAAAG")
Loc_AATACA_randomIntron= read.table("../data/SignalSiteFiles/Loc_AATACA_Distance2end_randomIntron.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AATACA")
Loc_AATAGA_randomIntron= read.table("../data/SignalSiteFiles/Loc_AATAGA_Distance2end_randomIntron.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AATAGA")
Loc_AATATA_randomIntron= read.table("../data/SignalSiteFiles/Loc_AATATA_Distance2end_randomIntron.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AATATA")
Loc_ACTAAA_randomIntron= read.table("../data/SignalSiteFiles/Loc_ACTAAA_Distance2end_randomIntron.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="ACTAAA")
Loc_AGTAAA_randomIntron= read.table("../data/SignalSiteFiles/Loc_AGTAAA_Distance2end_randomIntron.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AGTAAA")
Loc_ATTAAA_randomIntron= read.table("../data/SignalSiteFiles/Loc_ATTAAA_Distance2end_randomIntron.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="ATTAAA")
Loc_CATAAA_randomIntron= read.table("../data/SignalSiteFiles/Loc_CATAAA_Distance2end_randomIntron.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="CATAAA")
Loc_GATAAA_randomIntron= read.table("../data/SignalSiteFiles/Loc_GATAAA_Distance2end_randomIntron.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="GATAAA")
Loc_TATAAA_randomIntron= read.table("../data/SignalSiteFiles/Loc_TATAAA_Distance2end_randomIntron.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="TATAAA")
Loc_AAAAAA_randomIntron= read.table("../data/SignalSiteFiles/Loc_AAAAAA_Distance2end_randomIntron.txt", header=F, col.names =c( "PAS", "Distance2PAS")) %>% mutate(Site="AAAAAA")
```

Join these together:  

```{r}
AllsiteDF_randomIntron=as.data.frame(rbind(Loc_AATAAA_randomIntron,Loc_AAAAAG_randomIntron,Loc_AATACA_randomIntron,Loc_AATAGA_randomIntron,Loc_AATATA_randomIntron,Loc_ACTAAA_randomIntron,Loc_AGTAAA_randomIntron,Loc_ATTAAA_randomIntron, Loc_GATAAA_randomIntron,Loc_TATAAA_randomIntron,Loc_CATAAA_randomIntron, Loc_AAAAAA_randomIntron))
```

Number of tested sites: 


```{r}
withSS_random=nrow(AllsiteDF_randomIntron)
possiblereg=84432042

propwithRandom=withSS_random/possiblereg
propwithRandom
```


Difference in prop test:  

```{r}
prop.test(x=c(withSS_random,propwSS_intron), n=c(possiblereg,intronPAS),alternative = "less" )
```

