---
title: "SNPs in signal sites"
author: "Briana Mittleman"
date: "6/24/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this analysis I want to ask if snps in a signal site are more likely to be apaQTLs than other snps close to the PAS. In order to do this i need to subset to the pas that have signal site (identified [here](signalsiteanalysis.html)) I will then identyify the region 50 bp upstream of the PAS and ask if there are snps in this region using the vcf files for the snps i tested.  

```{r}
library(workflowr)
library(tidyverse)
library(ggpubr)
```

```{bash,eval=F}
mkdir ../data/SNPinSS
```


I want a bed file with 50bp upstream of these PAS. 
```{r}
PASwSS=read.table("../data/PAS/PASwSignalSite.txt", header = T,stringsAsFactors = F)
PAS=read.table("../data/PAS/APAPAS_GeneLocAnno.5perc.bed", stringsAsFactors = F, header = F, col.names = c("chr", "start", "end", "PASid","score", "strand")) %>% separate(PASid, into=c("pasNum", "geneiD"), sep=":") %>% mutate(PAS=paste("peak", pasNum, sep=""),PASname=paste(PAS, geneiD, sep="_")) 

PASwSSregion=PASwSS %>% inner_join(PAS, by="PAS") %>% mutate(newEnd=ifelse(strand=="+", end+50, end),newStart=ifelse(strand=="+", start, start-50)) %>% select(chr, newStart,newEnd, PASname, score, strand)

write.table(PASwSSregion,"../data/SNPinSS/FiftyupstreamPASwSS.bed", col.names = F, row.names = F, quote = F, sep="\t")

```


```{bash,eval=F}

sort -k1,1 -k2,2n ../data/SNPinSS/FiftyupstreamPASwSS.bed > ../data/SNPinSS/FiftyupstreamPASwSS.sortbed
sbatch subsetVCF_upstreamPAS.sh

cat  ../data/SNPinSS/SNPSinFiftyupstreamPAS_chr* >../data/SNPinSS/SNPSinFiftyupstreamPAS_Allchr.recode.vcf
```





I want to further subset to those in a signal site.  


```{r}
SSregions=PASwSS %>% inner_join(PAS, by="PAS") %>% mutate(absdist=abs(UpstreamDist),newEnd= ifelse(strand=="+", end-absdist, end+absdist), newStart=ifelse(strand=="+", end- (absdist+6), end + (absdist-6)), length=newEnd-newStart) %>% select(chr, newStart,newEnd, PASname, score, strand) 


write.table(SSregions,"../data/SNPinSS/SignalSiteRegions.bed", col.names = F, row.names = F, quote = F, sep="\t")
```

```{bash,eval=F}
../data/SNPinSS/SSregions_Allchr.sort.bed
sort -k1,1 -k2,2n ../data/SNPinSS/SignalSiteRegions.bed > ../data/SNPinSS/SignalSiteRegions.sort.bed
sbatch subsetVCF_SS.sh

cat  ../data/SNPinSS/SSregions_chr* > ../data/SNPinSS/SSregions_Allchr.recode.vcf
```


I will also need a different region to comare to. I can just shift these regions upstream by 7

```{r}
SS_diffregion=SSregions %>% mutate(randStart=ifelse(strand=="+", newStart-7, newEnd), randend=ifelse(strand=="+", newStart, newEnd+7), length=randend-randStart) %>% select(chr, randStart,randend, PASname, score, strand) 

write.table(SS_diffregion,"../data/SNPinSS/OtherSSRegions.bed", col.names = F, row.names = F, quote = F, sep="\t")
```


```{bash,eval=F}

sort -k1,1 -k2,2n ../data/SNPinSS/OtherSSRegions.bed  > ../data/SNPinSS/OtherSSRegions.sort.bed 
sbatch subsetvcf_otherreg.sh 

cat  ../data/SNPinSS/Otherregions_chr* > ../data/SNPinSS/Otherregions_Allchr.recode.vcf
```


Alternative option, permute the distances: 

```{r}
permdist=sample(PASwSS$UpstreamDist, length(PASwSS$UpstreamDist), replace = F)

SSregions_perm=as.data.frame(cbind(PASwSS,  permdist))%>% inner_join(PAS, by="PAS")  %>% mutate(absdist=abs(permdist),newEnd= ifelse(strand=="+", end-absdist, end+absdist), newStart=ifelse(strand=="+", end- (absdist+6), end + (absdist-6)), length=newEnd-newStart)%>% select(chr, newStart,newEnd, PASname, score, strand)

write.table(SSregions_perm,"../data/SNPinSS/SSRegions_permuted.bed", col.names = F, row.names = F, quote = F, sep="\t")
```


```{bash,eval=F}

sort -k1,1 -k2,2n ../data/SNPinSS/SSRegions_permuted.bed > ../data/SNPinSS/SSRegions_permuted.sort.bed
sbatch subsetvcf_permSS.sh 

cat  ../data/SNPinSS/SSRegionsPerm_chr* > ../data/SNPinSS/SSRegionsPerm_Allchr.recode.vcf
#remove # in first line
```


Pull in QTL snps:

```{r}
totQTLs=read.table("../data/apaQTLs/Total_apaQTLs4pc_5fdr.txt",header = T, stringsAsFactors = F) %>% select(sid) %>% unique()
write.table(totQTLs,"../data/apaQTLs/TotalQTLSNPsRSID.txt", col.names = F, row.names = F, quote = F)

nucQTLs=read.table("../data/apaQTLs/Nuclear_apaQTLs4pc_5fdr.txt",header = T, stringsAsFactors = F) %>% select(sid) %>% unique()
write.table(nucQTLs,"../data/apaQTLs/NuclearQTLSNPsRSID.txt", col.names = F, row.names = F, quote = F)
```


Signal site results:

```{r}
SS_snps=read.table("../data/SNPinSS/SSregions_Allchr.recode.vcf",header = T, stringsAsFactors = F) %>% select(ID) %>% mutate(totQTL=ifelse(ID %in% totQTLs$sid, "Yes", "No"), nucQTL=ifelse(ID %in% nucQTLs$sid, "Yes", "No"))


permutedSS_snps=read.table("../data/SNPinSS/SSRegionsPerm_Allchr.recode.vcf",header = T, stringsAsFactors = F) %>% select(ID) %>% mutate(totQTL=ifelse(ID %in% totQTLs$sid, "Yes", "No"), nucQTL=ifelse(ID %in% nucQTLs$sid, "Yes", "No"))

otherReg_snp=read.table("../data/SNPinSS/Otherregions_Allchr.recode.vcf",header = T, stringsAsFactors = F) %>% select(ID) %>% mutate(totQTL=ifelse(ID %in% totQTLs$sid, "Yes", "No"), nucQTL=ifelse(ID %in% nucQTLs$sid, "Yes", "No"))

fiftybp_snp=read.table("../data/SNPinSS/SNPSinFiftyupstreamPAS_Allchr.recode.vcf",header = T, stringsAsFactors = F) %>% select(ID) %>% mutate(totQTL=ifelse(ID %in% totQTLs$sid, "Yes", "No"), nucQTL=ifelse(ID %in% nucQTLs$sid, "Yes", "No"))
```

There are only 2 qtl snps in these signal sites. This is not enough to draw anything from this. 


Try with pvalues. Are the snps in pvals more likely to be significant than those not. 

I need to figure out which peak is associated with each snp.

I can make a bedfile from the SS snps in python and overlap this with the Signal site regions.  

```{bash,eval=F}
python vcf2bed.py ../data/SNPinSS/SSregions_Allchr.recode.vcf  ../data/SNPinSS/SSregions_Allchr.bed

sort -k1,1 -k2,2n ../data/SNPinSS/SSregions_Allchr.bed > ../data/SNPinSS/SSregions_Allchr.sort.bed


python vcf2bed.py ../data/SNPinSS/SSRegionsPerm_Allchr.recode.vcf  ../data/SNPinSS/SSRegionsPerm_Allchr.bed
sort -k1,1 -k2,2n ../data/SNPinSS/SSRegionsPerm_Allchr.bed> ../data/SNPinSS/SSRegionsPerm_Allchr.sort.bed



python vcf2bed.py ../data/SNPinSS/SNPSinFiftyupstreamPAS_Allchr.recode.vcf  ../data/SNPinSS/SNPSinFiftyupstreamPAS_Allchr.bed
sort -k1,1 -k2,2n ../data/SNPinSS/SNPSinFiftyupstreamPAS_Allchr.bed > ../data/SNPinSS/SNPSinFiftyupstreamPAS_Allchr.sort.bed


python vcf2bed.py ../data/SNPinSS/Otherregions_Allchr.recode.vcf  ../data/SNPinSS/Otherregions_Allchr.bed
sort -k1,1 -k2,2n ../data/SNPinSS/Otherregions_Allchr.bed > ../data/SNPinSS/Otherregions_Allchr.sort.bed


sort -k1,1 -k2,2n ../data/SNPinSS/SignalSiteRegions.bed > ../data/SNPinSS/SignalSiteRegions.sort.bed


sort -k1,1 -k2,2n ../data/SNPinSS/SSRegions_permuted.bed >../data/SNPinSS/SSRegions_permuted.sort.bed

sort -k1,1 -k2,2n ../data/SNPinSS/FiftyupstreamPASwSS.bed > ../data/SNPinSS/FiftyupstreamPASwSS.sort.bed

sort -k1,1 -k2,2n ../data/SNPinSS/OtherSSRegions.bed > ../data/SNPinSS/OtherSSRegions.sort.bed


```

intersect with bedtools to map the snps to the regions. Then I will be able to select the snp PAS associations.  
```{bash,eval=F}
sbatch mapSSsnps2PAS.sh
```

Results to get the associations:

```{r}
SSsnpswithPAS=read.table("../data/SNPinSS/SNPinSS2PAS.txt",col.names = c("chr","start", "end", "PASname", "score", "strand", "SNP")) %>% filter(SNP!=".") %>% separate(PASname, into=c("PAS", "gene", "loc"),sep="_") %>% select(PAS, SNP)
write.table(SSsnpswithPAS, "../data/SNPinSS/SS_PASandSNPs.txt", row.names = F, col.names = F, quote = F, sep="\t")


SSsnpswithPERMPAS=read.table("../data/SNPinSS/SNPinPermSS2PAS.txt",col.names = c("chr","start", "end", "PASname", "score", "strand", "SNP")) %>% filter(SNP!=".") %>% separate(PASname, into=c("PAS", "gene", "loc"),sep="_") %>% select(PAS, SNP)
write.table(SSsnpswithPERMPAS, "../data/SNPinSS/PermSS_PASandSNPs.txt", row.names = F, col.names = F, quote = F, sep="\t")

SNPregion=read.table("../data/SNPinSS/SNPSinFiftyupstream2PAS.txt",col.names = c("chr","start", "end", "PASname", "score", "strand", "SNP")) %>% filter(SNP!=".") %>% separate(PASname, into=c("PAS", "gene", "loc"),sep="_") %>% select(PAS, SNP)
write.table(SNPregion, "../data/SNPinSS/PASregion_PASandSNPs.txt", row.names = F, col.names = F, quote = F, sep="\t")


Otherregion=read.table("../data/SNPinSS/Otherregions2PAS.txt",col.names = c("chr","start", "end", "PASname", "score", "strand", "SNP")) %>% filter(SNP!=".") %>% separate(PASname, into=c("PAS", "gene", "loc"),sep="_") %>% select(PAS, SNP)
write.table(Otherregion, "../data/SNPinSS/Otherregions_PASandSNPs.txt", row.names = F, col.names = F, quote = F, sep="\t")
```

Process the snp region with python

```{bash,eval=F}
python fixPASregionSNPs.py

```

Run this with total and nuclear 
```{bash,eval=F}

sbatch RunRes2PAS.sh

#all of this code is in the above batch script 
python NomResfromPASSNP.py ../data/SNPinSS/SS_PASandSNPs.txt ../data/apaQTLNominal_4pc/APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz.qqnorm_AllChrom.txt ../data/apaQTLNominal_4pc/SS_Nuclear_nomRes.txt 

python NomResfromPASSNP.py ../data/SNPinSS/SS_PASandSNPs.txt ../data/apaQTLNominal_4pc/APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz.qqnorm_AllChrom.txt ../data/apaQTLNominal_4pc/SS_Total_nomRes.txt 


python NomResfromPASSNP.py ../data/SNPinSS/PermSS_PASandSNPs.txt ../data/apaQTLNominal_4pc/APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz.qqnorm_AllChrom.txt ../data/apaQTLNominal_4pc/PermSS_Nuclear_nomRes.txt 

python NomResfromPASSNP.py ../data/SNPinSS/PermSS_PASandSNPs.txt ../data/apaQTLNominal_4pc/APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz.qqnorm_AllChrom.txt ../data/apaQTLNominal_4pc/PermSS_Total_nomRes.txt 


python NomResfromPASSNP.py ../data/SNPinSS/PASregion_PASandSNPs.FIXED.txt ../data/apaQTLNominal_4pc/APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz.qqnorm_AllChrom.txt ../data/apaQTLNominal_4pc/RegionSS_Nuclear_nomRes.txt 

python NomResfromPASSNP.py ../data/SNPinSS/PASregion_PASandSNPs.FIXED.txt ../data/apaQTLNominal_4pc/APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz.qqnorm_AllChrom.txt ../data/apaQTLNominal_4pc/RegionSS_Total_nomRes.txt  

python NomResfromPASSNP.py ../data/SNPinSS/Otherregions_PASandSNPs.FIXED.txt ../data/apaQTLNominal_4pc/APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz.qqnorm_AllChrom.txt ../data/apaQTLNominal_4pc/OtherSS_Nuclear_nomRes.txt 

python NomResfromPASSNP.py ../data/SNPinSS/Otherregions_PASandSNPs.FIXED.txt ../data/apaQTLNominal_4pc/APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz.qqnorm_AllChrom.txt ../data/apaQTLNominal_4pc/OtherSS_Total_nomRes.txt  



```



```{r}
Nuclear_SS=read.table('../data/apaQTLNominal_4pc/SS_Nuclear_nomRes.txt', header = F, col.names=c("peakID", "snp", "dist", "pval", "slope"), stringsAsFactors = F) %>% select(pval,slope) %>% mutate(fraction="Nuclear", set="SS")
Nuclear_Perm=read.table('../data/apaQTLNominal_4pc/PermSS_Nuclear_nomRes.txt',header = F, col.names=c("peakID", "snp", "dist", "pval", "slope"), stringsAsFactors = F) %>% select(pval,slope) %>% mutate(fraction="Nuclear", set="Permuted")

Total_SS=read.table('../data/apaQTLNominal_4pc/SS_Total_nomRes.txt', header = F, col.names=c("peakID", "snp", "dist", "pval", "slope"), stringsAsFactors = F) %>% select(pval,slope) %>% mutate(fraction="Total", set="SS")
Total_Perm=read.table('../data/apaQTLNominal_4pc/PermSS_Total_nomRes.txt',header = F, col.names=c("peakID", "snp", "dist", "pval", "slope"), stringsAsFactors = F) %>% select(pval,slope) %>% mutate(fraction="Total", set="Permuted")


Nuclear_Region=read.table('../data/apaQTLNominal_4pc/RegionSS_Nuclear_nomRes.txt',header = F, col.names=c("peakID", "snp", "dist", "pval", "slope"), stringsAsFactors = F) %>% select(pval,slope) %>% mutate(fraction="Nuclear", set="Region")

Total_Region=read.table('../data/apaQTLNominal_4pc/RegionSS_Total_nomRes.txt',header = F, col.names=c("peakID", "snp", "dist", "pval", "slope"), stringsAsFactors = F) %>% select(pval,slope) %>% mutate(fraction="Total", set="Region")

Nuclear_other=read.table('../data/apaQTLNominal_4pc/OtherSS_Nuclear_nomRes.txt',header = F, col.names=c("peakID", "snp", "dist", "pval", "slope"), stringsAsFactors = F) %>% select(pval,slope) %>% mutate(fraction="Nuclear", set="Upstream")

Total_other=read.table('../data/apaQTLNominal_4pc/OtherSS_Total_nomRes.txt',header = F, col.names=c("peakID", "snp", "dist", "pval", "slope"), stringsAsFactors = F) %>% select(pval,slope) %>% mutate(fraction="Total", set="Upstream")


all_SS_pval=bind_rows(Nuclear_SS,Nuclear_Perm,Total_SS,Total_Perm,Nuclear_Region,Total_Region,Nuclear_other,Total_other)

all_SS_pval$set <- factor(all_SS_pval$set, levels=c("SS", "Upstream", "Permuted", "Region"))
```

```{r}
ggplot(all_SS_pval, aes(x=fraction, fill=set, y=pval)) + geom_boxplot() + labs(x="Fraction", title="p-values for SNPs in Signal Sites",y="nominal P-value" ) + scale_fill_discrete(name = 'Set', labels = c('Signal Sites', 'Region upstream of Signal Site', 'Permuted Distance to Signal Site',"50 bp upstream of PAS"))
```
```{r}
ggplot(all_SS_pval, aes(x=fraction, fill=set, y=abs(slope))) + geom_boxplot()  + labs(x="Fraction", title="p-values for SNPs in Signal Sites",y="absolute value effect size" ) + scale_fill_discrete(name = 'Set', labels = c('Signal Sites', 'Region upstream of Signal Site', 'Permuted Distance to Signal Site',"50 bp upstream of PAS"))
```

I am plotting this the wrong way. I need to make qqplots with the snps.  

```{r}
#plot qqplot
qqplot(-log10(runif(nrow(Nuclear_SS))), -log10(Nuclear_SS$pval))
points(sort(-log10(runif(nrow(Total_SS)))), sort(-log10(Total_SS$pval)),col= alpha("Red"))
points(sort(-log10(runif(nrow(Nuclear_Region)))), sort(-log10(Nuclear_Region$pval)),col= alpha("Orange"))
points(sort(-log10(runif(nrow(Total_Region)))), sort(-log10(Total_Region$pval)),col= alpha("Green"))

abline(0,1)

```

This is close but i need to make this a better comparison.I want just the UTR variants.  


```{r}
Nuclear_SS_all=read.table('../data/apaQTLNominal_4pc/SS_Nuclear_nomRes.txt', header = F, col.names=c("peakID", "snp", "dist", "pval", "slope"), stringsAsFactors = F)  %>% separate(peakID, into=c("chr", "start", "end", "PASid"), sep=":")%>% separate(PASid, into=c("gene", "loc","strand", "PAS"), sep="_") %>% filter(loc=="utr3")

Total_SS_all=read.table('../data/apaQTLNominal_4pc/SS_Total_nomRes.txt', header = F, col.names=c("peakID", "snp", "dist", "pval", "slope"), stringsAsFactors = F)%>% separate(peakID, into=c("chr", "start", "end", "PASid"), sep=":")%>% separate(PASid, into=c("gene", "loc","strand", "PAS"), sep="_") %>% filter(loc=="utr3")
```

The benchmark set is the UTRs without signal sites  

```{r}
PASwSS=read.table("../data/PAS/PASwSignalSite.txt", header = T,stringsAsFactors = F)
PAS_noSS=read.table("../data/PAS/APAPAS_GeneLocAnno.5perc.bed", stringsAsFactors = F, header = F, col.names = c("chr", "start", "end", "PASid","score", "strand")) %>% separate(PASid, into=c("pasNum", "geneiD"), sep=":") %>% mutate(PAS=paste("peak", pasNum, sep=""),PASname=paste(PAS, geneiD, sep="_")) %>% anti_join(PASwSS, by="PAS") %>% separate(geneiD,into=c("gene", "loc"), sep="_") %>% filter(loc=="utr3") %>% mutate(newEnd=ifelse(strand=="+", end+50, end),newStart=ifelse(strand=="+", start, start-50))%>% select(chr, newStart,newEnd, PASname, score, strand)

write.table(PAS_noSS,"../data/SNPinSS/UTRregionsPASnoSS.bed", col.names = F, row.names = F, quote = F, sep="\t")
```


```{bash,eval=F}

sort -k1,1 -k2,2n ../data/SNPinSS/UTRregionsPASnoSS.bed > ../data/SNPinSS/UTRregionsPASnoSS.sort.bed


sbatch subsetVCF_noSSregions.sh

cat  ../data/SNPinSS/UTRnoSS_chr* >../data/SNPinSS/UTRnoSS_Allchr.recode.vcf


python vcf2bed.py ../data/SNPinSS/UTRnoSS_Allchr.recode.vcf  ../data/SNPinSS/UTRnoSS_SNPsAllchr.bed

sort -k1,1 -k2,2n ../data/SNPinSS/UTRregionsPASnoSS.bed > ../data/SNPinSS/UTRregionsPASnoSS.sort.bed
sort -k1,1 -k2,2n ../data/SNPinSS/UTRnoSS_SNPsAllchr.bed > ../data/SNPinSS/UTRnoSS_SNPsAllchr.sort.bed

sbatch mapSSsnps2PAS.sh
```


```{r}
NOSSsnpswithPAS=read.table("../data/SNPinSS/UTRnoSS_SNPsAllchr2PAS.txt",col.names = c("chr","start", "end", "PASname", "score", "strand", "SNP")) %>% filter(SNP!=".") %>% separate(PASname, into=c("PAS", "gene", "loc"),sep="_") %>% select(PAS, SNP)
write.table(NOSSsnpswithPAS, "../data/SNPinSS/NoSSUTR_PASandSNPs.txt", row.names = F, col.names = F, quote = F, sep="\t")



```



Get the pval: 

```{bash,eval=F}
python NomResfromPASSNP.py ../data/SNPinSS/NoSSUTR_PASandSNPs.txt ../data/apaQTLNominal_4pc/APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz.qqnorm_AllChrom.txt ../data/apaQTLNominal_4pc/NoSSUTR_Nuclear_nomRes.txt 

python NomResfromPASSNP.py ../data/SNPinSS/NoSSUTR_PASandSNPs.txt ../data/apaQTLNominal_4pc/APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz.qqnorm_AllChrom.txt ../data/apaQTLNominal_4pc/NoSSUTR_Total_nomRes.txt 

```


```{r}
Nuclear_NOSS=read.table('../data/apaQTLNominal_4pc/NoSSUTR_Nuclear_nomRes.txt', header = F, col.names=c("peakID", "snp", "dist", "pval", "slope"), stringsAsFactors = F)  %>% separate(peakID, into=c("chr", "start", "end", "PASid"), sep=":")%>% separate(PASid, into=c("gene", "loc","strand", "PAS"), sep="_") %>% filter(loc=="utr3")

Total_NOSS=read.table('../data/apaQTLNominal_4pc/NoSSUTR_Total_nomRes.txt', header = F, col.names=c("peakID", "snp", "dist", "pval", "slope"), stringsAsFactors = F)%>% separate(peakID, into=c("chr", "start", "end", "PASid"), sep=":")%>% separate(PASid, into=c("gene", "loc","strand", "PAS"), sep="_") %>% filter(loc=="utr3")
```

Total
```{r}
#plot qqplot
qqplot(-log10(runif(nrow(Total_NOSS))), -log10(Total_NOSS$pval), xlab="-log10(Uniform)", ylab="-log10(pval)", main="Total Apa")
points(sort(-log10(runif(nrow(Total_SS)))), sort(-log10(Total_SS$pval)),col= alpha("Red"))
abline(0,1)
legend("topleft", legend=c("SNPs in UTR PAS Signal Sites", "SNPS not in Signal Sites"),col=c("red", "black"), pch=16,bty = 'n')

```
Nuclear:

```{r}
qqplot(-log10(runif(nrow(Nuclear_NOSS))), -log10(Nuclear_NOSS$pval),xlab="-log10(Uniform)", ylab="-log10(pval)", main="Nuclear Apa")
points(sort(-log10(runif(nrow(Nuclear_SS)))), sort(-log10(Nuclear_SS$pval)),col= alpha("Red"))
abline(0,1)

legend("topleft", legend=c("SNPs in UTR PAS Signal Sites", "SNPS not in Signal Sites"),col=c("red", "black"), pch=16,bty = 'n')

```


Assess significance:

```{r}
wilcox.test(Nuclear_SS$pval,Nuclear_NOSS$pval,alternative = 'less')
wilcox.test(Total_SS$pval,Total_NOSS$pval, alternative = "less")

```

